{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# DECON-EQTL PART2 : Running Decon-eQTL\n",
    "### (in biowulf)\n",
    "\n",
    "### Nov 13, 2020"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "\n",
    "cohort = 'nabec'\n",
    "date = 'nov2020'\n",
    "CHROMOSOMES=[str(x) for x in list(range(1,23))]\n",
    "#WRKDIR = f'/labshare/anni/eqtl/Decon-eQTL/{cohort}'\n",
    "WRKDIR = f'/data/LNG/anni/decon-eqtl/{cohort}'\n",
    "\n",
    "#expr_in = \n",
    "geno_dir = f'{WRKDIR}/genotypes'\n",
    "expr_dir = f'{WRKDIR}/expression'\n",
    "fract_in = f'{WRKDIR}'\n",
    "\n",
    "\n",
    "fract_file = 'nabec.scaden.v4_200.cell_fract.subset.txt'\n",
    "subset_samples_file = 'nabec.deconeqtl.subset.samples.txt'\n",
    "gencode_annots_file = f'{WRKDIR}/references/gencode.v32.transcripts.txt'\n",
    "genome_index_file = f'{WRKDIR}/references/Homo_sapiens_assembly38.fasta.fai'\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## generating sn file"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### setting up environment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(227462, 25)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqname</th>\n",
       "      <th>source</th>\n",
       "      <th>feature</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>frame</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>...</th>\n",
       "      <th>transcript_type</th>\n",
       "      <th>transcript_name</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>tag</th>\n",
       "      <th>havana_transcript</th>\n",
       "      <th>exon_number</th>\n",
       "      <th>exon_id</th>\n",
       "      <th>ont</th>\n",
       "      <th>protein_id</th>\n",
       "      <th>ccdsid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>transcript</td>\n",
       "      <td>11869</td>\n",
       "      <td>14409</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>ENSG00000223972.5</td>\n",
       "      <td>transcribed_unprocessed_pseudogene</td>\n",
       "      <td>...</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>DDX11L1-202</td>\n",
       "      <td>1.0</td>\n",
       "      <td>basic</td>\n",
       "      <td>OTTHUMT00000362751.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>transcript</td>\n",
       "      <td>12010</td>\n",
       "      <td>13670</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>ENSG00000223972.5</td>\n",
       "      <td>transcribed_unprocessed_pseudogene</td>\n",
       "      <td>...</td>\n",
       "      <td>transcribed_unprocessed_pseudogene</td>\n",
       "      <td>DDX11L1-201</td>\n",
       "      <td>NaN</td>\n",
       "      <td>basic</td>\n",
       "      <td>OTTHUMT00000002844.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PGO:0000005,PGO:0000019</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>transcript</td>\n",
       "      <td>14404</td>\n",
       "      <td>29570</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>0</td>\n",
       "      <td>ENSG00000227232.5</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td>...</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td>WASH7P-201</td>\n",
       "      <td>NaN</td>\n",
       "      <td>basic</td>\n",
       "      <td>OTTHUMT00000002839.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PGO:0000005</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>chr1</td>\n",
       "      <td>ENSEMBL</td>\n",
       "      <td>transcript</td>\n",
       "      <td>17369</td>\n",
       "      <td>17436</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>0</td>\n",
       "      <td>ENSG00000278267.1</td>\n",
       "      <td>miRNA</td>\n",
       "      <td>...</td>\n",
       "      <td>miRNA</td>\n",
       "      <td>MIR6859-1-201</td>\n",
       "      <td>NaN</td>\n",
       "      <td>basic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>transcript</td>\n",
       "      <td>29554</td>\n",
       "      <td>31097</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>ENSG00000243485.5</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>...</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>MIR1302-2HG-202</td>\n",
       "      <td>5.0</td>\n",
       "      <td>not_best_in_genome_evidence,dotter_confirmed,b...</td>\n",
       "      <td>OTTHUMT00000002840.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   seqname   source     feature  start    end  score strand  frame  \\\n",
       "1     chr1   HAVANA  transcript  11869  14409    NaN      +      0   \n",
       "5     chr1   HAVANA  transcript  12010  13670    NaN      +      0   \n",
       "13    chr1   HAVANA  transcript  14404  29570    NaN      -      0   \n",
       "26    chr1  ENSEMBL  transcript  17369  17436    NaN      -      0   \n",
       "29    chr1   HAVANA  transcript  29554  31097    NaN      +      0   \n",
       "\n",
       "              gene_id                           gene_type  ...  \\\n",
       "1   ENSG00000223972.5  transcribed_unprocessed_pseudogene  ...   \n",
       "5   ENSG00000223972.5  transcribed_unprocessed_pseudogene  ...   \n",
       "13  ENSG00000227232.5              unprocessed_pseudogene  ...   \n",
       "26  ENSG00000278267.1                               miRNA  ...   \n",
       "29  ENSG00000243485.5                              lncRNA  ...   \n",
       "\n",
       "                       transcript_type  transcript_name  \\\n",
       "1                               lncRNA      DDX11L1-202   \n",
       "5   transcribed_unprocessed_pseudogene      DDX11L1-201   \n",
       "13              unprocessed_pseudogene       WASH7P-201   \n",
       "26                               miRNA    MIR6859-1-201   \n",
       "29                              lncRNA  MIR1302-2HG-202   \n",
       "\n",
       "   transcript_support_level  \\\n",
       "1                       1.0   \n",
       "5                       NaN   \n",
       "13                      NaN   \n",
       "26                      NaN   \n",
       "29                      5.0   \n",
       "\n",
       "                                                  tag     havana_transcript  \\\n",
       "1                                               basic  OTTHUMT00000362751.1   \n",
       "5                                               basic  OTTHUMT00000002844.2   \n",
       "13                                              basic  OTTHUMT00000002839.1   \n",
       "26                                              basic                   NaN   \n",
       "29  not_best_in_genome_evidence,dotter_confirmed,b...  OTTHUMT00000002840.1   \n",
       "\n",
       "   exon_number exon_id                      ont protein_id ccdsid  \n",
       "1          NaN     NaN                      NaN        NaN    NaN  \n",
       "5          NaN     NaN  PGO:0000005,PGO:0000019        NaN    NaN  \n",
       "13         NaN     NaN              PGO:0000005        NaN    NaN  \n",
       "26         NaN     NaN                      NaN        NaN    NaN  \n",
       "29         NaN     NaN                      NaN        NaN    NaN  \n",
       "\n",
       "[5 rows x 25 columns]"
      ]
     },
     "execution_count": 88,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#read in the transcript annotations from txt\n",
    "#gencode_annots_file = WRKDIR + '/gencode.v32.transcripts.txt.gz'\n",
    "\n",
    "gencode = pd.read_csv(gencode_annots_file,sep='\\t',comment='#', index_col=0)\n",
    "print(gencode.shape)\n",
    "gencode.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### prep scripts to generate gene/snp list per chrom"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "run in terminal: module load plink/2.0-dev-20190318\n"
     ]
    }
   ],
   "source": [
    "print('run in terminal: module load plink/2.0-dev-20190318')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1675 transcripts detected on chromosome 1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/mooreank/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:34: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "/home/mooreank/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:35: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1185 transcripts detected on chromosome 2\n",
      "1024 transcripts detected on chromosome 3\n",
      "671 transcripts detected on chromosome 4\n",
      "853 transcripts detected on chromosome 5\n",
      "880 transcripts detected on chromosome 6\n",
      "853 transcripts detected on chromosome 7\n",
      "647 transcripts detected on chromosome 8\n",
      "700 transcripts detected on chromosome 9\n",
      "646 transcripts detected on chromosome 10\n",
      "898 transcripts detected on chromosome 11\n",
      "951 transcripts detected on chromosome 12\n",
      "334 transcripts detected on chromosome 13\n",
      "570 transcripts detected on chromosome 14\n",
      "610 transcripts detected on chromosome 15\n",
      "777 transcripts detected on chromosome 16\n",
      "1001 transcripts detected on chromosome 17\n",
      "288 transcripts detected on chromosome 18\n",
      "1114 transcripts detected on chromosome 19\n",
      "415 transcripts detected on chromosome 20\n",
      "194 transcripts detected on chromosome 21\n",
      "402 transcripts detected on chromosome 22\n",
      "total genes: 16688\n"
     ]
    }
   ],
   "source": [
    "##section modified from raph's ppmi_wd_eqtl.ipynb script\n",
    "\n",
    "#run plink to get all SNPs for each gene determined to be on chrom based on start/end sites\n",
    "#combine all gene/SNP output file to one list\n",
    "!mkdir /data/LNG/anni/decon-eqtl/nabec/plink\n",
    "!mkdir /data/LNG/anni/decon-eqtl/nabec/plink/output\n",
    "!mkdir /data/LNG/anni/decon-eqtl/nabec/plink/scripts\n",
    "!mkdir /data/LNG/anni/decon-eqtl/nabec/plink/output/gene_snps\n",
    "!mkdir /data/LNG/anni/decon-eqtl/nabec/plink/output/temp\n",
    "\n",
    "\n",
    "#chrom=21\n",
    "total_transcripts = 0\n",
    "for chrom in CHROMOSOMES:\n",
    "\n",
    "    exp_df = pd.read_csv(f'{WRKDIR}/expression/nabec.aug2020.chr{chrom}.subset.pheno', sep='\\t', index_col=0)\n",
    "\n",
    "\n",
    "    #genome_index_file = f'{WRKDIR}/Homo_sapiens_assembly38.fasta.fai'\n",
    "    chr_lengths_df = pd.read_csv(genome_index_file, header=None, sep='\\t')\n",
    "\n",
    "    max_dist = 1000000\n",
    "\n",
    "    #for chrom in CHROMOSOMES:\n",
    "    #chr_genes = gencode.loc[gencode['seqid'] == 'chr{}'.format(chrom)]\n",
    "    chr_genes = gencode.loc[gencode['seqname'] == 'chr{}'.format(chrom)]\n",
    "\n",
    "    chr_genes_id = set(chr_genes['gene_id']) & set(exp_df.index)\n",
    "    print('{} transcripts detected on chromosome {}'.format(len(chr_genes_id),chrom))\n",
    "    total_transcripts += int(len(chr_genes_id))\n",
    "    chr_genes_df = chr_genes.loc[chr_genes['gene_id'].isin(chr_genes_id)]\n",
    "\n",
    "    #compute the +/- cis proximal start and stop\n",
    "    chr_genes_df['cisstart'] = chr_genes_df['start'] - max_dist\n",
    "    chr_genes_df['cisstop'] = chr_genes_df['end'] + max_dist    \n",
    "\n",
    "    #clean up if cis start or end is beyond chromosome beginning or end\n",
    "    chr_genes_df.loc[chr_genes_df['cisstart'] < 0, 'cisstart'] = 0\n",
    "    max_end = chr_lengths_df.loc[chr_lengths_df[0] == 'chr{}'.format(chrom),1].values[0]\n",
    "    chr_genes_df.loc[chr_genes_df['cisstop'] > max_end, 'cisstop'] = max_end\n",
    "\n",
    "    chr_genes_df = chr_genes_df[['gene_id', 'seqname', 'start', 'end', 'strand', \\\n",
    "                                            'cisstart', 'cisstop', 'gene_name']]\n",
    "\n",
    "    # chr_detected_out_file = '{}/plink/output/snp/detected.genes.chr{}'.format(WRKDIR,chrom)\n",
    "    # chr_genes_df.to_csv(chr_detected_out_file, sep='\\t', index=False, header=True)\n",
    "\n",
    "    #now format the biowulf job files\n",
    "    command_plink2 = 'plink --silent --pfile {A_WRKDIR}/genotypes/nabec.aug2020.chr{A_CHROM}.geno05maf05hwe000001 \\\n",
    "    --chr chr{A_CHROM} --from-bp {A_FROMBP} --to-bp {A_TOBP} \\\n",
    "    --pheno-name {A_PHENONAME} \\\n",
    "    --keep {A_WRKDIR}/{A_SUBFILE} \\\n",
    "    --make-pgen \\\n",
    "    --out {A_WRKDIR}/plink/output/temp/{A_PHENONAME}\\n'\n",
    "\n",
    "\n",
    "    OUTHEADER='\\\"GENEID\\\\tSNPID\\\"'\n",
    "    #     cmd_accumulate = 'awk -v OFS=\"'\"\\\\t\"'\" '\"'/^#CHROM/{{ r=NR }}r && NR>r {{ print \"'\"{A_PHENONAME}\"'\",\"'\"chr\"'\"$1\"'\":\"'\"$2\"'\":\"'\"$4\"'\":\"'\"$5 }}'\"' \\\n",
    "    #     {A_WRKDIR}/plink/output/temp/{A_VISIT}/chr{A_CHROM}/{A_PHENONAME}.pvar >> {A_WRKDIR}/plink/output/snp/{A_VISIT}.genes.chr{A_CHROM}.plink.snp.list\\n'\n",
    "    cmd_echo_header = 'echo -e {A_HEADER} > {A_WRKDIR}/plink/output/gene_snps/chr{A_CHROM}.{A_PHENONAME}.plink.snp.list\\n'\n",
    "    cmd_format = 'awk -v OFS=\"'\"\\\\t\"'\" '\"'/^#CHROM/{{ r=NR }}r && NR>r {{ print \"'\"{A_PHENONAME}\"'\",\"'\"chr\"'\"$1\"'\":\"'\"$2\"'\":\"'\"$4\"'\":\"'\"$5 }}'\"' \\\n",
    "    {A_WRKDIR}/plink/output/temp/{A_PHENONAME}.pvar >> {A_WRKDIR}/plink/output/gene_snps/chr{A_CHROM}.{A_PHENONAME}.plink.snp.list\\n'\n",
    "\n",
    "    cmd_rm_result = 'rm {A_WRKDIR}/plink/output/temp/{A_PHENONAME}*\\n'\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "    #cmd_gzip = 'gzip {A_WRKDIR}/plink/output/snp/{A_VISIT}.genes.chr{A_CHROM}.plink.snp.list\\n'\n",
    "\n",
    "\n",
    "    #generate chr job file for each visit\n",
    "    #for visit in visits_oi:\n",
    "    chr_job_file_name = f'{WRKDIR}/plink/scripts/nabec.snp.list.chr{chrom}.jobs'       \n",
    "    with open(chr_job_file_name,'w') as chr_job_file:                                    \n",
    "        for gene_id in chr_genes_id:\n",
    "            gene_transcripts_df =  chr_genes_df.loc[chr_genes_df['gene_id'] == gene_id]\n",
    "            cis_start = min(gene_transcripts_df['cisstart'])\n",
    "            cos_stop = max(gene_transcripts_df['cisstop'])    \n",
    "            chr_job_file.write(command_plink2.format(A_WRKDIR=WRKDIR, \\\n",
    "                                                     A_CHROM=chrom, \\\n",
    "                                                     A_PHENONAME=gene_id, A_FROMBP=cis_start, \\\n",
    "                                                     A_TOBP=cos_stop, A_SUBFILE = subset_samples_file))\n",
    "            chr_job_file.write(cmd_echo_header.format(A_HEADER=OUTHEADER, \\\n",
    "                                                      A_WRKDIR=WRKDIR, \\\n",
    "                                                      A_PHENONAME=gene_id, A_CHROM=chrom)) \n",
    "            chr_job_file.write(cmd_format.format(A_PHENONAME=gene_id, A_WRKDIR=WRKDIR, \\\n",
    "                                                     A_CHROM=chrom))\n",
    "            chr_job_file.write(cmd_rm_result.format(A_WRKDIR=WRKDIR, \\\n",
    "                                                    A_PHENONAME=gene_id))        \n",
    "        #now output cmds for zipping up chr files\n",
    "        #chr_job_file.write(cmd_gzip.format(A_WRKDIR=WRKDIR, A_VISIT=visit, A_CHROM=chrom))\n",
    "        \n",
    "print(f'total genes: '+str(total_transcripts))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [],
   "source": [
    "#format the swarm jobfile(s)\n",
    "#for visit in visits_oi:\n",
    "swarm_file_name = f'{WRKDIR}/plink/scripts/run_nabec_gene_snps.swarm'\n",
    "\n",
    "with open(swarm_file_name,'w') as swarm_file:\n",
    "    for chrom in CHROMOSOMES:\n",
    "        chr_job_file_name = f'{WRKDIR}/plink/scripts/nabec.snp.list.chr{chrom}.jobs\\n' \n",
    "        swarm_file.write(chr_job_file_name)\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### run scripts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "#for chrom in CHROMOSOMES:\n",
    "!mkdir /data/LNG/anni/decon-eqtl/ppmi/plink/output/temp/V08\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#run these commands at terminal:\n",
      "\n",
      "chmod +x /data/LNG/anni/decon-eqtl/ppmi/plink/scripts/*\n",
      "swarm -g 64 -t 24 -f run_ppmi_gene_snps.V08.swarm \n"
     ]
    }
   ],
   "source": [
    "#format and make the swarm files executable\n",
    "print('#run these commands at terminal:\\n')\n",
    "print(f'chmod +x {WRKDIR}/plink/scripts/*')\n",
    "#for chrom in CHROMOSOMES:\n",
    "SWARMFILENAME='{}/plink/scripts/run_ppmi_wb_eqtl.V08.swarm'.format(WRKDIR,visit)\n",
    "#CHRFILENAME = '{}/plink/scripts/ppmi.snp.list.chr{}.jobs'.format(WRKDIR,chrom)\n",
    "#LOGFILENAME='{}/plink/output/temp/run_ppmi_wb_eqtl.V08.log'.format(WRKDIR,chrom)\n",
    "print(f'swarm -g 64 -t 24 -f run_ppmi_gene_snps.{visit}.swarm ')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26986\n"
     ]
    }
   ],
   "source": [
    "## check number of gene files created matches earlier number\n",
    "cmd = f'ls /data/LNG/anni/decon-eqtl/ppmi/plink/output/gene_snps/ | wc -l '\n",
    "!{cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GENEID</th>\n",
       "      <th>SNPID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000238284.2</td>\n",
       "      <td>chr7:41661765:C:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000238284.2</td>\n",
       "      <td>chr7:41662670:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000238284.2</td>\n",
       "      <td>chr7:41662736:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000238284.2</td>\n",
       "      <td>chr7:41663058:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000238284.2</td>\n",
       "      <td>chr7:41663704:A:C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              GENEID              SNPID\n",
       "0  ENSG00000238284.2  chr7:41661765:C:A\n",
       "1  ENSG00000238284.2  chr7:41662670:G:A\n",
       "2  ENSG00000238284.2  chr7:41662736:C:T\n",
       "3  ENSG00000238284.2  chr7:41663058:G:A\n",
       "4  ENSG00000238284.2  chr7:41663704:A:C"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## check one\n",
    "test = pd.read_csv(f'{WRKDIR}/plink/output/gene_snps/V08.ENSG00000238284.2.plink.snp.list', sep='\\t')\n",
    "test.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cat ./temp/chr1/*.csv > ppmi.deconeqtl.results.chr1.cat.csv\n",
      "cat ./temp/chr2/*.csv > ppmi.deconeqtl.results.chr2.cat.csv\n",
      "cat ./temp/chr3/*.csv > ppmi.deconeqtl.results.chr3.cat.csv\n",
      "cat ./temp/chr4/*.csv > ppmi.deconeqtl.results.chr4.cat.csv\n",
      "cat ./temp/chr5/*.csv > ppmi.deconeqtl.results.chr5.cat.csv\n",
      "cat ./temp/chr6/*.csv > ppmi.deconeqtl.results.chr6.cat.csv\n",
      "cat ./temp/chr7/*.csv > ppmi.deconeqtl.results.chr7.cat.csv\n",
      "cat ./temp/chr8/*.csv > ppmi.deconeqtl.results.chr8.cat.csv\n",
      "cat ./temp/chr9/*.csv > ppmi.deconeqtl.results.chr9.cat.csv\n",
      "cat ./temp/chr10/*.csv > ppmi.deconeqtl.results.chr10.cat.csv\n",
      "cat ./temp/chr11/*.csv > ppmi.deconeqtl.results.chr11.cat.csv\n",
      "cat ./temp/chr12/*.csv > ppmi.deconeqtl.results.chr12.cat.csv\n",
      "cat ./temp/chr13/*.csv > ppmi.deconeqtl.results.chr13.cat.csv\n",
      "cat ./temp/chr14/*.csv > ppmi.deconeqtl.results.chr14.cat.csv\n",
      "cat ./temp/chr15/*.csv > ppmi.deconeqtl.results.chr15.cat.csv\n",
      "cat ./temp/chr16/*.csv > ppmi.deconeqtl.results.chr16.cat.csv\n",
      "cat ./temp/chr17/*.csv > ppmi.deconeqtl.results.chr17.cat.csv\n",
      "cat ./temp/chr18/*.csv > ppmi.deconeqtl.results.chr18.cat.csv\n",
      "cat ./temp/chr19/*.csv > ppmi.deconeqtl.results.chr19.cat.csv\n",
      "cat ./temp/chr20/*.csv > ppmi.deconeqtl.results.chr20.cat.csv\n",
      "cat ./temp/chr21/*.csv > ppmi.deconeqtl.results.chr21.cat.csv\n",
      "cat ./temp/chr22/*.csv > ppmi.deconeqtl.results.chr22.cat.csv\n"
     ]
    }
   ],
   "source": [
    "for x in CHROMOSOMES:\n",
    "    print(f'cat ./temp/chr{x}/*.csv > ppmi.deconeqtl.results.chr{x}.cat.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### delete all individual gene files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "# cmd_rm_result = f'rm {WRKDIR}/plink/output/temp/{visit}/chr{chrom}/*'\n",
    "# !{cmd_rm_result}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "# ##checking output file\n",
    "# snp_df = pd.read_csv(f'{WRKDIR}/plink/output/snp/V08.genes.chr22.plink.snp.list', sep='\\t')\n",
    "# print(snp_df.shape)\n",
    "# snp_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### run command in terminal (within directory containing decon jar file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for chrom in CHROMOSOMES:\n",
    "    cmd=f'mkdir {WRKDIR}/results/temp/chr{chrom}'\n",
    "    #print(cmd)\n",
    "    !{cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "for chrom in CHROMOSOMES:\n",
    "    cmd=f'mkdir {WRKDIR}/scripts/chr{chrom}'\n",
    "    #print(cmd)\n",
    "    !{cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "####job file per gene, swarm file per chrom\n",
    "\n",
    "#read in the transcript annotations from txt\n",
    "#gencode_annots_file = WRKDIR + '/gencode.v32.transcripts.txt.gz'\n",
    "#gencode_annots_file = f'{WRKDIR}/gencode.v32.transcripts.txt.gz'\n",
    "gencode = pd.read_csv(gencode_annots_file,sep='\\t',comment='#', index_col=0)\n",
    "\n",
    "\n",
    "## get gene list for eahc chrom again\n",
    "#CHROMOSOMES = [22]\n",
    "for chrom in CHROMOSOMES:\n",
    "    chr_result_file_name = f'{WRKDIR}/results/{cohort}.deconeqtl.chr{chrom}.results.csv'\n",
    "    with open(chr_result_file_name,'w') as chr_result_file: \n",
    "        header = '\\tBasophils_pvalue\\tEosinophils_pvalue\\tLymphocytes_pvalue\\tMonocytes_pvalue\\tNeutrophils_pvalue\\tBeta1_Basophils\\tBeta2_Eosinophils\\tBeta3_Lymphocytes\\tBeta4_Monocytes\\tBeta5_Neutrophils\tBeta6_Basophils:GT\\tBeta7_Eosinophils:GT\\tBeta8_Lymphocytes:GT\\tBeta9_Monocytes:GT\\tBeta10_Neutrophils:GT'\n",
    "        cmd_echo_header = f'{header} > {chr_result_file}\\n'\n",
    "        chr_result_file.write(cmd_echo_header)\n",
    "        \n",
    "\n",
    "    exp_df = pd.read_csv(f'{WRKDIR}/expression/{cohort}.aug2020.chr{chrom}.subset.pheno', sep='\\t', index_col=0)\n",
    "\n",
    "\n",
    "    #genome_index_file = f'{WRKDIR}/Homo_sapiens_assembly38.fasta.fai'\n",
    "    chr_lengths_df = pd.read_csv(genome_index_file, header=None, sep='\\t')\n",
    "\n",
    "    max_dist = 1000000\n",
    "\n",
    "    #for chrom in CHROMOSOMES:\n",
    "    #chr_genes = gencode.loc[gencode['seqid'] == 'chr{}'.format(chrom)]\n",
    "    chr_genes = gencode.loc[gencode['seqname'] == 'chr{}'.format(chrom)]\n",
    "\n",
    "    chr_genes_id = set(chr_genes['gene_id']) & set(exp_df.index)\n",
    "    print('{} transcripts detected on chromosome {}'.format(len(chr_genes_id),chrom))\n",
    "    total_transcripts += int(len(chr_genes_id))\n",
    "    chr_genes_df = chr_genes.loc[chr_genes['gene_id'].isin(chr_genes_id)]\n",
    "\n",
    "    #compute the +/- cis proximal start and stop\n",
    "    chr_genes_df['cisstart'] = chr_genes_df['start'] - max_dist\n",
    "    chr_genes_df['cisstop'] = chr_genes_df['end'] + max_dist    \n",
    "\n",
    "    #clean up if cis start or end is beyond chromosome beginning or end\n",
    "    chr_genes_df.loc[chr_genes_df['cisstart'] < 0, 'cisstart'] = 0\n",
    "    max_end = chr_lengths_df.loc[chr_lengths_df[0] == 'chr{}'.format(chrom),1].values[0]\n",
    "    chr_genes_df.loc[chr_genes_df['cisstop'] > max_end, 'cisstop'] = max_end\n",
    "\n",
    "    chr_genes_df = chr_genes_df[['gene_id', 'seqname', 'start', 'end', 'strand', \\\n",
    "                                            'cisstart', 'cisstop', 'gene_name']]\n",
    "    \n",
    "    ##for each gene write decon-eqtl command, add result to total chrom results file\n",
    "    for gene_id in chr_genes_id:\n",
    "        \n",
    "        gene_job_file_name = f'{WRKDIR}/scripts/chr{chrom}/{cohort}.deconeqtl.{gene_id}.job' \n",
    "    \n",
    "        with open(gene_job_file_name,'w') as gene_job_file: \n",
    "#             \n",
    "\n",
    "            #gene_output_file = f'{WRKDIR}/output/V08.ppmi.deconResults.sept2020.chr{chrom}.csv'  \n",
    "\n",
    "            gene_transcripts_df =  chr_genes_df.loc[chr_genes_df['gene_id'] == gene_id]\n",
    "\n",
    "            #now format the biowulf job files\n",
    "            command_java = f'java -Xmx100g \\\n",
    "                        -jar {WRKDIR}/Decon-eQTL-v*.*.*-jar-with-dependencies.jar \\\n",
    "                        -c {WRKDIR}/{fract_file} \\\n",
    "                        -e {WRKDIR}/expression/{cohort}.aug2020.chr{chrom}.subset.pheno\\\n",
    "                        -g {WRKDIR}/genotypes/{cohort}.aug2020.geno05maf05hwe000001.chr{chrom}.subset.traw \\\n",
    "                        --no_console \\\n",
    "                        -o {WRKDIR}/results/temp/chr{chrom} \\\n",
    "                        -of {cohort}.deconResults.{date}.{gene_id}.csv \\\n",
    "                        -sn {WRKDIR}/plink/output/gene_snps/chr{chrom}.{gene_id}.plink.snp.list\\n \\\n",
    "                        > {WRKDIR}/logs/deconvolutionResults.{date}.{gene_id}.log\\n'\n",
    "\n",
    "\n",
    "            command_accum = f'awk '\"'NR>1 {{ print }} '\"' {}/results/temp/chr{}/{}.deconResults.{}.{}.csv >> {}\\n'.format(WRKDIR,chrom,cohort,date,gene_id,chr_result_file_name)\n",
    "            #command_rmv_snp = f'rm {WRKDIR}/output/gene_snps/V08.{gene}.chr{chrom}.plink.snp.list'\n",
    "            command_rmv_result = f'rm {WRKDIR}/results/temp/chr{chrom}/{cohort}.deconResults.{date}.{gene_id}.csv'\n",
    "\n",
    "            gene_job_file.write(command_java) \n",
    "\n",
    "            gene_job_file.write(command_accum)\n",
    "            \n",
    "    #format the swarm jobfile(s)\n",
    "    ##collect all gene jobs file names to one file for swarm\n",
    "    swarm_file_name = f'{WRKDIR}/scripts/run_{cohort}_deconeqtl_chr{chrom}.swarm'\n",
    "    with open(swarm_file_name,'w') as swarm_file:\n",
    "        for gene_id in chr_genes_id:\n",
    "            gene_job_file_name = f'{WRKDIR}/scripts/chr{chrom}/{cohort}.deconeqtl.{gene_id}.job\\n' \n",
    "            swarm_file.write(gene_job_file_name)\n",
    "\n",
    "\n",
    "     \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#run these commands at terminal in Decon directory:\n",
      "\n",
      "chmod +x /data/LNG/anni/decon-eqtl/ppmi/scripts/chr1/*\n",
      "chmod +x /data/LNG/anni/decon-eqtl/ppmi/results/temp/chr1\n",
      "swarm -g 15 -t 4 -f /data/LNG/anni/decon-eqtl/ppmi/scripts/run_ppmi_V08_deconeqtl_chr1.swarm\n",
      "chmod +x /data/LNG/anni/decon-eqtl/ppmi/scripts/chr2/*\n",
      "chmod +x /data/LNG/anni/decon-eqtl/ppmi/results/temp/chr2\n",
      "swarm -g 15 -t 4 -f /data/LNG/anni/decon-eqtl/ppmi/scripts/run_ppmi_V08_deconeqtl_chr2.swarm\n",
      "chmod +x /data/LNG/anni/decon-eqtl/ppmi/scripts/chr3/*\n",
      "chmod +x /data/LNG/anni/decon-eqtl/ppmi/results/temp/chr3\n",
      "swarm -g 15 -t 4 -f /data/LNG/anni/decon-eqtl/ppmi/scripts/run_ppmi_V08_deconeqtl_chr3.swarm\n",
      "chmod +x /data/LNG/anni/decon-eqtl/ppmi/scripts/chr4/*\n",
      "chmod +x /data/LNG/anni/decon-eqtl/ppmi/results/temp/chr4\n",
      "swarm -g 15 -t 4 -f /data/LNG/anni/decon-eqtl/ppmi/scripts/run_ppmi_V08_deconeqtl_chr4.swarm\n",
      "chmod +x /data/LNG/anni/decon-eqtl/ppmi/scripts/chr5/*\n",
      "chmod +x /data/LNG/anni/decon-eqtl/ppmi/results/temp/chr5\n",
      "swarm -g 15 -t 4 -f /data/LNG/anni/decon-eqtl/ppmi/scripts/run_ppmi_V08_deconeqtl_chr5.swarm\n",
      "chmod +x /data/LNG/anni/decon-eqtl/ppmi/scripts/chr6/*\n",
      "chmod +x /data/LNG/anni/decon-eqtl/ppmi/results/temp/chr6\n",
      "swarm -g 15 -t 4 -f /data/LNG/anni/decon-eqtl/ppmi/scripts/run_ppmi_V08_deconeqtl_chr6.swarm\n",
      "chmod +x /data/LNG/anni/decon-eqtl/ppmi/scripts/chr7/*\n",
      "chmod +x /data/LNG/anni/decon-eqtl/ppmi/results/temp/chr7\n",
      "swarm -g 15 -t 4 -f /data/LNG/anni/decon-eqtl/ppmi/scripts/run_ppmi_V08_deconeqtl_chr7.swarm\n",
      "chmod +x /data/LNG/anni/decon-eqtl/ppmi/scripts/chr8/*\n",
      "chmod +x /data/LNG/anni/decon-eqtl/ppmi/results/temp/chr8\n",
      "swarm -g 15 -t 4 -f /data/LNG/anni/decon-eqtl/ppmi/scripts/run_ppmi_V08_deconeqtl_chr8.swarm\n",
      "chmod +x /data/LNG/anni/decon-eqtl/ppmi/scripts/chr9/*\n",
      "chmod +x /data/LNG/anni/decon-eqtl/ppmi/results/temp/chr9\n",
      "swarm -g 15 -t 4 -f /data/LNG/anni/decon-eqtl/ppmi/scripts/run_ppmi_V08_deconeqtl_chr9.swarm\n",
      "chmod +x /data/LNG/anni/decon-eqtl/ppmi/scripts/chr10/*\n",
      "chmod +x /data/LNG/anni/decon-eqtl/ppmi/results/temp/chr10\n",
      "swarm -g 15 -t 4 -f /data/LNG/anni/decon-eqtl/ppmi/scripts/run_ppmi_V08_deconeqtl_chr10.swarm\n",
      "chmod +x /data/LNG/anni/decon-eqtl/ppmi/scripts/chr11/*\n",
      "chmod +x /data/LNG/anni/decon-eqtl/ppmi/results/temp/chr11\n",
      "swarm -g 15 -t 4 -f /data/LNG/anni/decon-eqtl/ppmi/scripts/run_ppmi_V08_deconeqtl_chr11.swarm\n",
      "chmod +x /data/LNG/anni/decon-eqtl/ppmi/scripts/chr12/*\n",
      "chmod +x /data/LNG/anni/decon-eqtl/ppmi/results/temp/chr12\n",
      "swarm -g 15 -t 4 -f /data/LNG/anni/decon-eqtl/ppmi/scripts/run_ppmi_V08_deconeqtl_chr12.swarm\n",
      "chmod +x /data/LNG/anni/decon-eqtl/ppmi/scripts/chr13/*\n",
      "chmod +x /data/LNG/anni/decon-eqtl/ppmi/results/temp/chr13\n",
      "swarm -g 15 -t 4 -f /data/LNG/anni/decon-eqtl/ppmi/scripts/run_ppmi_V08_deconeqtl_chr13.swarm\n",
      "chmod +x /data/LNG/anni/decon-eqtl/ppmi/scripts/chr14/*\n",
      "chmod +x /data/LNG/anni/decon-eqtl/ppmi/results/temp/chr14\n",
      "swarm -g 15 -t 4 -f /data/LNG/anni/decon-eqtl/ppmi/scripts/run_ppmi_V08_deconeqtl_chr14.swarm\n",
      "chmod +x /data/LNG/anni/decon-eqtl/ppmi/scripts/chr15/*\n",
      "chmod +x /data/LNG/anni/decon-eqtl/ppmi/results/temp/chr15\n",
      "swarm -g 15 -t 4 -f /data/LNG/anni/decon-eqtl/ppmi/scripts/run_ppmi_V08_deconeqtl_chr15.swarm\n",
      "chmod +x /data/LNG/anni/decon-eqtl/ppmi/scripts/chr16/*\n",
      "chmod +x /data/LNG/anni/decon-eqtl/ppmi/results/temp/chr16\n",
      "swarm -g 15 -t 4 -f /data/LNG/anni/decon-eqtl/ppmi/scripts/run_ppmi_V08_deconeqtl_chr16.swarm\n",
      "chmod +x /data/LNG/anni/decon-eqtl/ppmi/scripts/chr17/*\n",
      "chmod +x /data/LNG/anni/decon-eqtl/ppmi/results/temp/chr17\n",
      "swarm -g 15 -t 4 -f /data/LNG/anni/decon-eqtl/ppmi/scripts/run_ppmi_V08_deconeqtl_chr17.swarm\n",
      "chmod +x /data/LNG/anni/decon-eqtl/ppmi/scripts/chr18/*\n",
      "chmod +x /data/LNG/anni/decon-eqtl/ppmi/results/temp/chr18\n",
      "swarm -g 15 -t 4 -f /data/LNG/anni/decon-eqtl/ppmi/scripts/run_ppmi_V08_deconeqtl_chr18.swarm\n",
      "chmod +x /data/LNG/anni/decon-eqtl/ppmi/scripts/chr19/*\n",
      "chmod +x /data/LNG/anni/decon-eqtl/ppmi/results/temp/chr19\n",
      "swarm -g 15 -t 4 -f /data/LNG/anni/decon-eqtl/ppmi/scripts/run_ppmi_V08_deconeqtl_chr19.swarm\n",
      "chmod +x /data/LNG/anni/decon-eqtl/ppmi/scripts/chr20/*\n",
      "chmod +x /data/LNG/anni/decon-eqtl/ppmi/results/temp/chr20\n",
      "swarm -g 15 -t 4 -f /data/LNG/anni/decon-eqtl/ppmi/scripts/run_ppmi_V08_deconeqtl_chr20.swarm\n",
      "chmod +x /data/LNG/anni/decon-eqtl/ppmi/scripts/chr21/*\n",
      "chmod +x /data/LNG/anni/decon-eqtl/ppmi/results/temp/chr21\n",
      "swarm -g 15 -t 4 -f /data/LNG/anni/decon-eqtl/ppmi/scripts/run_ppmi_V08_deconeqtl_chr21.swarm\n",
      "chmod +x /data/LNG/anni/decon-eqtl/ppmi/scripts/chr22/*\n",
      "chmod +x /data/LNG/anni/decon-eqtl/ppmi/results/temp/chr22\n",
      "swarm -g 15 -t 4 -f /data/LNG/anni/decon-eqtl/ppmi/scripts/run_ppmi_V08_deconeqtl_chr22.swarm\n"
     ]
    }
   ],
   "source": [
    "#format and make the swarm files executable\n",
    "visit='V08'\n",
    "## run swarm file for each chrom\n",
    "print('#run these commands at terminal in Decon directory:\\n')\n",
    "#print(f'chmod +x {WRKDIR}/scripts/*')\n",
    "for chrom in CHROMOSOMES:\n",
    "    print(f'chmod +x {WRKDIR}/scripts/chr{chrom}/*')\n",
    "    print(f'chmod +x {WRKDIR}/results/temp/chr{chrom}')\n",
    "    #for visit in visits_oi:\n",
    "    SWARMCHRFILENAME=f'{WRKDIR}/scripts/run_{cohort}_deconeqtl_chr{chrom}.swarm'\n",
    "    #LOGFILENAME=f'{WRKDIR}/output/temp/ppmi.run.pairs.chr{chrom}.log'\n",
    "    print(f'swarm -g 15 -t 4 -f {SWARMCHRFILENAME}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## all results will be per chrom\n",
    "## need to load each chrom, merge\n",
    "\n",
    "result_df = pd.read_csv(f'{WRKDIR}/results/{cohort}.deconeqtl.chr1.results.csv', sep = '\\t')\n",
    "print('chrom1: '+str(result_df.shape))\n",
    "\n",
    "def load_result(chrom, return_df):\n",
    "    next_df = pd.read_csv(f'{WRKDIR}/results/{cohort}.deconeqtl.chr{chrom}.results.csv', sep = '\\t')\n",
    "    result_df = result_df.merge(next_df)\n",
    "    print(f'chrom{chrom}: '+str(result_df.shape))\n",
    "    return result_df\n",
    "\n",
    "CHROMOSOMES_sub=[str(x) for x in list(range(2,23))]\n",
    "for chrom in CHROMOSOMES_sub:\n",
    "    result_df = load_result(chrom, result_df)\n",
    "    \n",
    "result_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
