{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# DECON-EQTL PART3: Looking at results\n",
    "\n",
    "## Looking at Decon-eQTL results in ppmi blood bulk expression data with cell fractions\n",
    "\n",
    "### Novemeber 18, 2020\n",
    "\n",
    "#### https://github.com/molgenis/systemsgenetics/tree/master/Decon2/Decon-eQTL\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Set up environment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from rpy2.robjects.packages import importr\n",
    "#from rpy2.robjects.vectors import FloatVector\n",
    "import h5py\n",
    "\n",
    "stats = importr('stats')\n",
    "#qvalue = importr('qvalue')\n",
    "\n",
    "\n",
    "cohort = 'nabec'\n",
    "WRKDIR = f'/labshare/anni/eqtl/Decon-eQTL/{cohort}'\n",
    "CHROMOSOMES = [str(x) for x in list(range(1,23))]\n",
    "CHROMOSOMES_sub = [str(x) for x in list(range(2,23))]\n",
    "cell_types = ['Microglia','Astrocyte','OPC','Oligodendrocyte','Neuron']\n",
    "cell1 = 'Microglia'\n",
    "cell2 = 'Astrocyte'\n",
    "cell3 = 'OPC'\n",
    "cell4 = 'Oligodendrocyte'\n",
    "cell5 = 'Neuron'\n",
    "\n",
    "\n",
    "outdir = f'/labshare/anni/eqtl/Decon-eQTL/{cohort}/results'\n",
    "indir = f'/labshare/anni/eqtl/Decon-eQTL/{cohort}/results'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "##functions\n",
    "\n",
    "def find_signif(cell_type):\n",
    "    sig_df = results_df.loc[results_df[f'{cell_type}_bh_fdr'] < 0.05]\n",
    "    print(f'{cell_type}: '+str(sig_df.shape))\n",
    "    return sig_df\n",
    "\n",
    "def find_nonsignif(cell_type):\n",
    "    sig_df = results_df.loc[results_df[f'{cell_type}_bh_fdr'] > 0.05]\n",
    "    print(f'{cell_type}: '+str(sig_df.shape))\n",
    "    return sig_df\n",
    "\n",
    "def find_nonsignif_genes(cell_type_sig_genes, cell_type):\n",
    "    nonsig_genes = set(all_genes_decon) - set(cell_type_sig_genes)\n",
    "    print(f'{cell_type}: '+str(len(nonsig_genes)))\n",
    "    return nonsig_genes\n",
    "    \n",
    "def to_list(sig_df):\n",
    "    sig_vars = sig_df.index.tolist()\n",
    "    sig_genes = list(set(sig_df['GENEID'].tolist()))\n",
    "    print('significant genes: '+str(len(sig_genes)))\n",
    "    return sig_vars,sig_genes\n",
    "\n",
    "    \n",
    "def to_list_non(sig_df):\n",
    "    sig_vars = sig_df.index.tolist()\n",
    "    sig_genes = list(set(sig_df['GENEID'].tolist()))\n",
    "    print('nonsignificant genes: '+str(len(sig_genes)))\n",
    "    return sig_vars,sig_genes\n",
    "\n",
    "def compare(list1, list2, cell_type):\n",
    "    overlap = list(set(list1) & set(list2))\n",
    "    print(f'{cell_type}: '+str(len(overlap)))\n",
    "    return overlap\n",
    "\n",
    "def compare_plinknonsig(cell_list, cell_type):\n",
    "    overlap = list(set(cell_list) - set(plink_sig_genes))\n",
    "    print(f'{cell_type}: '+str(len(overlap)))\n",
    "    return overlap\n",
    "\n",
    "def genelist_to_file(gene_list, file_name):\n",
    "    with open(f'{outdir}/{file_name}', 'w') as f:\n",
    "        for item in gene_list:\n",
    "            f.write(\"%s\\n\" % item)\n",
    "            \n",
    "def file_to_list(filename, cell_type):\n",
    "    list_name = [line.rstrip('\\n') for line in open(f'{indir}/{filename}')]\n",
    "    print(f'{cell_type}: '+str(len(list_name)))\n",
    "    return list_name\n",
    "\n",
    "def p_adj(cell_type, in_df):\n",
    "    in_df[f'{cell_type}_pvalue'] = in_df[f'{cell_type}_pvalue'].astype(float)\n",
    "\n",
    "    ##adjust p value\n",
    "    bhAdj = stats.p_adjust((in_df[f'{cell_type}_pvalue'].tolist()), method = 'BH')\n",
    "    in_df[f'{cell_type}_bh_fdr'] = bhAdj\n",
    "    print(in_df.shape)\n",
    "    return in_df\n",
    "\n",
    "def calc_fdr(chrom):\n",
    "    results_df = pd.read_csv(f'{WRKDIR}/{cohort}.deconResults.nov2020.chr{chrom}.csv', sep='\\t', index_col=0, header=None)\n",
    "    results_df.columns = ['Microglia_pvalue','Astrocyte_pvalue','OPC_pvalue','Oligodendrocyte_pvalue','Neuron_pvalue','Beta1_Microglia','Beta2_Astrocyte','Beta3_OPC','Beta4_Oligodendrocyte','Beta5_Neuron','Beta6_Microglia:GT','Beta7_Astrocyte:GT','Beta8_OPC:GT','Beta9_Oligodendrocyte:GT','Beta10_Neuron:GT']\n",
    "    results_df['VAR'] = results_df.index\n",
    "    results_df['GENEID'] = results_df['VAR'].str.split(\"_\").str[0]\n",
    "    # for cell in cell_types:\n",
    "    #     results_df = p_adj(cell, results_df)\n",
    "    print(f'chrom{chrom}: '+str(results_df.shape))\n",
    "    for cell in cell_types:\n",
    "        results_df = p_adj(cell, results_df)\n",
    "    results_df.to_csv(f'{WRKDIR}/{cohort}.deconResults.nov2020.chr{chrom}.fdr.csv',sep='\\t',index=True)\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for x in CHROMOSOMES:\n",
    "    calc_fdr(x)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Format uploaded biowulf files \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## move to results location to manipulate\n",
    "# !cp /labshare/anni/eqtl/Decon-eQTL/ppmi/from_biowulf /labshare/anni/eqtl/Decon-eQTL/ppmi/results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# ## format biowulf files\n",
    "# ##stripping out any headers that also got concatenated\n",
    "# for chrom in CHROMOSOMES:\n",
    "#     cmd = 'awk '\"'BEGIN {{ OFS=FS=\"'\"\\\\t\"'\" }} $2 !~ /^Microglia/'\"' {}.deconeqtl.chr{}.results.csv > {}.deconeqtl.chr{}.results.strip.csv &'.format(cohort,chrom,cohort, chrom)\n",
    "#     print(cmd)\n",
    "    \n",
    "# ##concat all into one file\n",
    "# cmd = f'cat ./*strip.csv > {cohort}.deconeqtl.results.all.cat.csv'  \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## filter for just significant variants for each cell type\n",
    "## subset for significant into separate files\n",
    "\n",
    "# awk -F\"\\t\" '$2<0.05 {print}' ppmi.deconeqtl.results.all.cat.csv > ppmi.deconeqtl.results.all.cat.basophils_sig.csv &\n",
    "# awk -F\"\\t\" '$3<0.05 {print}' ppmi.deconeqtl.results.all.cat.csv > ppmi.deconeqtl.results.all.cat.eosinophils_sig.csv &\n",
    "# awk -F\"\\t\" '$4<0.05 {print}' ppmi.deconeqtl.results.all.cat.csv > ppmi.deconeqtl.results.all.cat.lymphocytes_sig.csv &\n",
    "# awk -F\"\\t\" '$5<0.05 {print}' ppmi.deconeqtl.results.all.cat.csv > ppmi.deconeqtl.results.all.cat.monocytes_sig.csv &\n",
    "# awk -F\"\\t\" '$6<0.05 {print}' ppmi.deconeqtl.results.all.cat.csv > ppmi.deconeqtl.results.all.cat.neutrophils_sig.csv &\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Correcting p values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cp /labshare/anni/notebooks/eqtl/ppmi/a_adj.by_chrom.py /labshare/anni/notebooks/eqtl/nabec/\n"
     ]
    }
   ],
   "source": [
    "#print(f'cp /labshare/anni/notebooks/eqtl/ppmi/a_adj.by_chrom.py /labshare/anni/notebooks/eqtl/{cohort}/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chrom1: (6360474, 17)\n",
      "(6360474, 18)\n",
      "(6360474, 19)\n",
      "(6360474, 20)\n",
      "(6360474, 21)\n",
      "(6360474, 22)\n",
      "chrom2: (4902388, 17)\n",
      "(4902388, 18)\n",
      "(4902388, 19)\n",
      "(4902388, 20)\n",
      "(4902388, 21)\n",
      "(4902388, 22)\n",
      "chrom3: (4366689, 17)\n",
      "(4366689, 18)\n",
      "(4366689, 19)\n",
      "(4366689, 20)\n",
      "(4366689, 21)\n",
      "(4366689, 22)\n",
      "chrom4: (3249307, 17)\n",
      "(3249307, 18)\n",
      "(3249307, 19)\n",
      "(3249307, 20)\n",
      "(3249307, 21)\n",
      "(3249307, 22)\n",
      "chrom5: (3581520, 17)\n",
      "(3581520, 18)\n",
      "(3581520, 19)\n",
      "(3581520, 20)\n",
      "(3581520, 21)\n",
      "(3581520, 22)\n",
      "chrom6: (5128387, 17)\n",
      "(5128387, 18)\n",
      "(5128387, 19)\n",
      "(5128387, 20)\n",
      "(5128387, 21)\n",
      "(5128387, 22)\n",
      "chrom7: (3767901, 17)\n",
      "(3767901, 18)\n",
      "(3767901, 19)\n",
      "(3767901, 20)\n",
      "(3767901, 21)\n",
      "(3767901, 22)\n",
      "chrom8: (2875022, 17)\n",
      "(2875022, 18)\n",
      "(2875022, 19)\n",
      "(2875022, 20)\n",
      "(2875022, 21)\n",
      "(2875022, 22)\n",
      "chrom9: (2775991, 17)\n",
      "(2775991, 18)\n",
      "(2775991, 19)\n",
      "(2775991, 20)\n",
      "(2775991, 21)\n",
      "(2775991, 22)\n",
      "chrom10: (2905647, 17)\n",
      "(2905647, 18)\n",
      "(2905647, 19)\n",
      "(2905647, 20)\n",
      "(2905647, 21)\n",
      "(2905647, 22)\n",
      "chrom11: (3797409, 17)\n",
      "(3797409, 18)\n",
      "(3797409, 19)\n",
      "(3797409, 20)\n",
      "(3797409, 21)\n",
      "(3797409, 22)\n",
      "chrom12: (4141852, 17)\n",
      "(4141852, 18)\n",
      "(4141852, 19)\n",
      "(4141852, 20)\n",
      "(4141852, 21)\n",
      "(4141852, 22)\n",
      "chrom13: (1588529, 17)\n",
      "(1588529, 18)\n",
      "(1588529, 19)\n",
      "(1588529, 20)\n",
      "(1588529, 21)\n",
      "(1588529, 22)\n",
      "chrom14: (2576250, 17)\n",
      "(2576250, 18)\n",
      "(2576250, 19)\n",
      "(2576250, 20)\n",
      "(2576250, 21)\n",
      "(2576250, 22)\n",
      "chrom15: (2400680, 17)\n",
      "(2400680, 18)\n",
      "(2400680, 19)\n",
      "(2400680, 20)\n",
      "(2400680, 21)\n",
      "(2400680, 22)\n",
      "chrom16: (3085349, 17)\n",
      "(3085349, 18)\n",
      "(3085349, 19)\n",
      "(3085349, 20)\n",
      "(3085349, 21)\n",
      "(3085349, 22)\n",
      "chrom17: (3941616, 17)\n",
      "(3941616, 18)\n",
      "(3941616, 19)\n",
      "(3941616, 20)\n",
      "(3941616, 21)\n",
      "(3941616, 22)\n",
      "chrom18: (1360111, 17)\n",
      "(1360111, 18)\n",
      "(1360111, 19)\n",
      "(1360111, 20)\n",
      "(1360111, 21)\n",
      "(1360111, 22)\n",
      "chrom19: (5125461, 17)\n",
      "(5125461, 18)\n",
      "(5125461, 19)\n",
      "(5125461, 20)\n",
      "(5125461, 21)\n",
      "(5125461, 22)\n",
      "chrom20: (1789977, 17)\n",
      "(1789977, 18)\n",
      "(1789977, 19)\n",
      "(1789977, 20)\n",
      "(1789977, 21)\n",
      "(1789977, 22)\n",
      "chrom21: (849087, 17)\n",
      "(849087, 18)\n",
      "(849087, 19)\n",
      "(849087, 20)\n",
      "(849087, 21)\n",
      "(849087, 22)\n",
      "chrom22: (1671619, 17)\n",
      "(1671619, 18)\n",
      "(1671619, 19)\n",
      "(1671619, 20)\n",
      "(1671619, 21)\n",
      "(1671619, 22)\n"
     ]
    }
   ],
   "source": [
    "## /labshare/anni/notebooks/eqtl/nabec/nabec_a_adj.by_chrom.py\n",
    "\n",
    "import pandas as pd\n",
    "from rpy2.robjects.packages import importr\n",
    "#from rpy2.robjects.vectors import FloatVector\n",
    "\n",
    "\n",
    "\n",
    "stats = importr('stats')\n",
    "#qvalue = importr('qvalue')\n",
    "\n",
    "\n",
    "cohort = 'nabec'\n",
    "WRKDIR = f'/labshare/anni/eqtl/Decon-eQTL/{cohort}/results_biowulf'\n",
    "CHROMOSOMES = [str(x) for x in list(range(1,23))]\n",
    "CHROMOSOMES_sub = [str(x) for x in list(range(2,23))]\n",
    "cell_types = ['Microglia','Astrocyte','OPC','Oligodendrocyte','Neuron']\n",
    "\n",
    "def p_adj(cell_type, in_df):\n",
    "    in_df[f'{cell_type}_pvalue'] = in_df[f'{cell_type}_pvalue'].astype(float)\n",
    "\n",
    "    ##adjust p value\n",
    "    bhAdj = stats.p_adjust((in_df[f'{cell_type}_pvalue'].tolist()), method = 'BH')\n",
    "    in_df[f'{cell_type}_bh_fdr'] = bhAdj\n",
    "    print(in_df.shape)\n",
    "    return in_df\n",
    "\n",
    "def calc_fdr(chrom):\n",
    "    results_df = pd.read_csv(f'{WRKDIR}/{cohort}.deconResults.nov2020.chr{chrom}.csv', sep='\\t', index_col=0, header=None)\n",
    "    results_df.columns = ['Microglia_pvalue','Astrocyte_pvalue','OPC_pvalue','Oligodendrocyte_pvalue','Neuron_pvalue','Beta1_Microglia','Beta2_Astrocyte','Beta3_OPC','Beta4_Oligodendrocyte','Beta5_Neuron','Beta6_Microglia:GT','Beta7_Astrocyte:GT','Beta8_OPC:GT','Beta9_Oligodendrocyte:GT','Beta10_Neuron:GT']\n",
    "    results_df['VAR'] = results_df.index\n",
    "    results_df['GENEID'] = results_df['VAR'].str.split(\"_\").str[0]\n",
    "    # for cell in cell_types:\n",
    "    #     results_df = p_adj(cell, results_df)\n",
    "    print(f'chrom{chrom}: '+str(results_df.shape))\n",
    "    for cell in cell_types:\n",
    "        results_df = p_adj(cell, results_df)\n",
    "    results_df.to_csv(f'{WRKDIR}/{cohort}.deconResults.nov2020.chr{chrom}.fdr.csv',sep='\\t',index=True)\n",
    "\n",
    "    \n",
    "\n",
    "for x in CHROMOSOMES:\n",
    "    calc_fdr(x)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Load in all chrom results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# #first 6 columns are the p-values for each of the cell types, and the last 6 columns are the beta (effect size) of the cell type - \n",
    "# #genotype interaction effect, relative to the allele that is coded as 2 in the dosage file.\n",
    "\n",
    "# results_df = pd.read_csv(f'{WRKDIR}/ppmi.deconeqtl.results.chr1.cat.strip.csv', sep='\\t', index_col=0)\n",
    "# results_df.columns = ['Basophils_pvalue','Eosinophils_pvalue','Lymphocytes_pvalue','Monocytes_pvalue','Neutrophils_pvalue','Beta1_Basophils','Beta2_Eosinophils','Beta3_Lymphocytes','Beta4_Monocytes','Beta5_Neutrophils','Beta6_Basophils:GT','Beta7_Eosinophils:GT','Beta8_Lymphocytes:GT','Beta9_Monocytes:GT','Beta10_Neutrophils:GT']\n",
    "# results_df['VAR'] = results_df.index\n",
    "# results_df['GENEID'] = results_df['VAR'].str.split(\"_\").str[0]\n",
    "# # for cell in cell_types:\n",
    "# #     results_df = p_adj(cell, results_df)\n",
    "# print('chrom1: '+str(results_df.shape))\n",
    "\n",
    "# for chrom in CHROMOSOMES_sub:\n",
    "#     next_df = pd.read_csv(f'{WRKDIR}/ppmi.deconeqtl.results.chr{chrom}.cat.strip.csv', sep='\\t', index_col=0)\n",
    "#     next_df.columns = ['Basophils_pvalue','Eosinophils_pvalue','Lymphocytes_pvalue','Monocytes_pvalue','Neutrophils_pvalue','Beta1_Basophils','Beta2_Eosinophils','Beta3_Lymphocytes','Beta4_Monocytes','Beta5_Neutrophils','Beta6_Basophils:GT','Beta7_Eosinophils:GT','Beta8_Lymphocytes:GT','Beta9_Monocytes:GT','Beta10_Neutrophils:GT']\n",
    "#     next_df['VAR'] = next_df.index\n",
    "#     next_df['GENEID'] = next_df['VAR'].str.split(\"_\").str[0]\n",
    "# #     for cell in cell_types:\n",
    "# #         results_df = p_adj(cell, results_df)\n",
    "#     results_df = pd.concat([results_df,next_df])\n",
    "#     print(f'plus chrom{chrom}: '+str(results_df.shape))\n",
    "\n",
    "   \n",
    "# results_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chrom1: (6360474, 22)\n",
      "plus chrom2: (11262862, 22)\n",
      "plus chrom3: (15629551, 22)\n",
      "plus chrom4: (18878858, 22)\n",
      "plus chrom5: (22460378, 22)\n",
      "plus chrom6: (27588765, 22)\n",
      "plus chrom7: (31356666, 22)\n",
      "plus chrom8: (34231688, 22)\n",
      "plus chrom9: (37007679, 22)\n",
      "plus chrom10: (39913326, 22)\n",
      "plus chrom11: (43710735, 22)\n",
      "plus chrom12: (47852587, 22)\n",
      "plus chrom13: (49441116, 22)\n",
      "plus chrom14: (52017366, 22)\n",
      "plus chrom15: (54418046, 22)\n",
      "plus chrom16: (57503395, 22)\n",
      "plus chrom17: (61445011, 22)\n",
      "plus chrom18: (62805122, 22)\n",
      "plus chrom19: (67930583, 22)\n",
      "plus chrom20: (69720560, 22)\n",
      "plus chrom21: (70569647, 22)\n",
      "plus chrom22: (72241266, 22)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Microglia_pvalue</th>\n",
       "      <th>Astrocyte_pvalue</th>\n",
       "      <th>OPC_pvalue</th>\n",
       "      <th>Oligodendrocyte_pvalue</th>\n",
       "      <th>Neuron_pvalue</th>\n",
       "      <th>Beta1_Microglia</th>\n",
       "      <th>Beta2_Astrocyte</th>\n",
       "      <th>Beta3_OPC</th>\n",
       "      <th>Beta4_Oligodendrocyte</th>\n",
       "      <th>Beta5_Neuron</th>\n",
       "      <th>...</th>\n",
       "      <th>Beta8_OPC:GT</th>\n",
       "      <th>Beta9_Oligodendrocyte:GT</th>\n",
       "      <th>Beta10_Neuron:GT</th>\n",
       "      <th>VAR</th>\n",
       "      <th>GENEID</th>\n",
       "      <th>Microglia_bh_fdr</th>\n",
       "      <th>Astrocyte_bh_fdr</th>\n",
       "      <th>OPC_bh_fdr</th>\n",
       "      <th>Oligodendrocyte_bh_fdr</th>\n",
       "      <th>Neuron_bh_fdr</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSG00000000457.14_chr1:168849631:C:A</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.358177</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.001498</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.015554</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>ENSG00000000457.14_chr1:168849631:C:A</td>\n",
       "      <td>ENSG00000000457.14</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000000457.14_chr1:168851510:G:A</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.549740</td>\n",
       "      <td>0.830599</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.011772</td>\n",
       "      <td>-0.000231</td>\n",
       "      <td>ENSG00000000457.14_chr1:168851510:G:A</td>\n",
       "      <td>ENSG00000000457.14</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000000457.14_chr1:168851541:T:G</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.549740</td>\n",
       "      <td>0.830599</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.011772</td>\n",
       "      <td>-0.000231</td>\n",
       "      <td>ENSG00000000457.14_chr1:168851541:T:G</td>\n",
       "      <td>ENSG00000000457.14</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000000457.14_chr1:168851651:G:A</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.552327</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.002907</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.010761</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>ENSG00000000457.14_chr1:168851651:G:A</td>\n",
       "      <td>ENSG00000000457.14</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000000457.14_chr1:168852658:C:T</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.640277</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.005355</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.009515</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>ENSG00000000457.14_chr1:168852658:C:T</td>\n",
       "      <td>ENSG00000000457.14</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       Microglia_pvalue  Astrocyte_pvalue  \\\n",
       "0                                                                           \n",
       "ENSG00000000457.14_chr1:168849631:C:A               1.0               1.0   \n",
       "ENSG00000000457.14_chr1:168851510:G:A               1.0               1.0   \n",
       "ENSG00000000457.14_chr1:168851541:T:G               1.0               1.0   \n",
       "ENSG00000000457.14_chr1:168851651:G:A               1.0               1.0   \n",
       "ENSG00000000457.14_chr1:168852658:C:T               1.0               1.0   \n",
       "\n",
       "                                       OPC_pvalue  Oligodendrocyte_pvalue  \\\n",
       "0                                                                           \n",
       "ENSG00000000457.14_chr1:168849631:C:A         1.0                0.358177   \n",
       "ENSG00000000457.14_chr1:168851510:G:A         1.0                0.549740   \n",
       "ENSG00000000457.14_chr1:168851541:T:G         1.0                0.549740   \n",
       "ENSG00000000457.14_chr1:168851651:G:A         1.0                0.552327   \n",
       "ENSG00000000457.14_chr1:168852658:C:T         1.0                0.640277   \n",
       "\n",
       "                                       Neuron_pvalue  Beta1_Microglia  \\\n",
       "0                                                                       \n",
       "ENSG00000000457.14_chr1:168849631:C:A       1.000000              0.0   \n",
       "ENSG00000000457.14_chr1:168851510:G:A       0.830599              0.0   \n",
       "ENSG00000000457.14_chr1:168851541:T:G       0.830599              0.0   \n",
       "ENSG00000000457.14_chr1:168851651:G:A       1.000000              0.0   \n",
       "ENSG00000000457.14_chr1:168852658:C:T       1.000000              0.0   \n",
       "\n",
       "                                       Beta2_Astrocyte  Beta3_OPC  \\\n",
       "0                                                                   \n",
       "ENSG00000000457.14_chr1:168849631:C:A              0.0        0.0   \n",
       "ENSG00000000457.14_chr1:168851510:G:A              0.0        0.0   \n",
       "ENSG00000000457.14_chr1:168851541:T:G              0.0        0.0   \n",
       "ENSG00000000457.14_chr1:168851651:G:A              0.0        0.0   \n",
       "ENSG00000000457.14_chr1:168852658:C:T              0.0        0.0   \n",
       "\n",
       "                                       Beta4_Oligodendrocyte  Beta5_Neuron  \\\n",
       "0                                                                            \n",
       "ENSG00000000457.14_chr1:168849631:C:A               0.001498           0.0   \n",
       "ENSG00000000457.14_chr1:168851510:G:A               0.000000           0.0   \n",
       "ENSG00000000457.14_chr1:168851541:T:G               0.000000           0.0   \n",
       "ENSG00000000457.14_chr1:168851651:G:A               0.002907           0.0   \n",
       "ENSG00000000457.14_chr1:168852658:C:T               0.005355           0.0   \n",
       "\n",
       "                                       ...  Beta8_OPC:GT  \\\n",
       "0                                      ...                 \n",
       "ENSG00000000457.14_chr1:168849631:C:A  ...           0.0   \n",
       "ENSG00000000457.14_chr1:168851510:G:A  ...           0.0   \n",
       "ENSG00000000457.14_chr1:168851541:T:G  ...           0.0   \n",
       "ENSG00000000457.14_chr1:168851651:G:A  ...           0.0   \n",
       "ENSG00000000457.14_chr1:168852658:C:T  ...           0.0   \n",
       "\n",
       "                                       Beta9_Oligodendrocyte:GT  \\\n",
       "0                                                                 \n",
       "ENSG00000000457.14_chr1:168849631:C:A                 -0.015554   \n",
       "ENSG00000000457.14_chr1:168851510:G:A                  0.011772   \n",
       "ENSG00000000457.14_chr1:168851541:T:G                  0.011772   \n",
       "ENSG00000000457.14_chr1:168851651:G:A                  0.010761   \n",
       "ENSG00000000457.14_chr1:168852658:C:T                 -0.009515   \n",
       "\n",
       "                                       Beta10_Neuron:GT  \\\n",
       "0                                                         \n",
       "ENSG00000000457.14_chr1:168849631:C:A          0.000000   \n",
       "ENSG00000000457.14_chr1:168851510:G:A         -0.000231   \n",
       "ENSG00000000457.14_chr1:168851541:T:G         -0.000231   \n",
       "ENSG00000000457.14_chr1:168851651:G:A          0.000000   \n",
       "ENSG00000000457.14_chr1:168852658:C:T          0.000000   \n",
       "\n",
       "                                                                         VAR  \\\n",
       "0                                                                              \n",
       "ENSG00000000457.14_chr1:168849631:C:A  ENSG00000000457.14_chr1:168849631:C:A   \n",
       "ENSG00000000457.14_chr1:168851510:G:A  ENSG00000000457.14_chr1:168851510:G:A   \n",
       "ENSG00000000457.14_chr1:168851541:T:G  ENSG00000000457.14_chr1:168851541:T:G   \n",
       "ENSG00000000457.14_chr1:168851651:G:A  ENSG00000000457.14_chr1:168851651:G:A   \n",
       "ENSG00000000457.14_chr1:168852658:C:T  ENSG00000000457.14_chr1:168852658:C:T   \n",
       "\n",
       "                                                   GENEID Microglia_bh_fdr  \\\n",
       "0                                                                            \n",
       "ENSG00000000457.14_chr1:168849631:C:A  ENSG00000000457.14              1.0   \n",
       "ENSG00000000457.14_chr1:168851510:G:A  ENSG00000000457.14              1.0   \n",
       "ENSG00000000457.14_chr1:168851541:T:G  ENSG00000000457.14              1.0   \n",
       "ENSG00000000457.14_chr1:168851651:G:A  ENSG00000000457.14              1.0   \n",
       "ENSG00000000457.14_chr1:168852658:C:T  ENSG00000000457.14              1.0   \n",
       "\n",
       "                                      Astrocyte_bh_fdr  OPC_bh_fdr  \\\n",
       "0                                                                    \n",
       "ENSG00000000457.14_chr1:168849631:C:A              1.0         1.0   \n",
       "ENSG00000000457.14_chr1:168851510:G:A              1.0         1.0   \n",
       "ENSG00000000457.14_chr1:168851541:T:G              1.0         1.0   \n",
       "ENSG00000000457.14_chr1:168851651:G:A              1.0         1.0   \n",
       "ENSG00000000457.14_chr1:168852658:C:T              1.0         1.0   \n",
       "\n",
       "                                       Oligodendrocyte_bh_fdr  Neuron_bh_fdr  \n",
       "0                                                                             \n",
       "ENSG00000000457.14_chr1:168849631:C:A                     1.0            1.0  \n",
       "ENSG00000000457.14_chr1:168851510:G:A                     1.0            1.0  \n",
       "ENSG00000000457.14_chr1:168851541:T:G                     1.0            1.0  \n",
       "ENSG00000000457.14_chr1:168851651:G:A                     1.0            1.0  \n",
       "ENSG00000000457.14_chr1:168852658:C:T                     1.0            1.0  \n",
       "\n",
       "[5 rows x 22 columns]"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#first 6 columns are the p-values for each of the cell types, and the last 6 columns are the beta (effect size) of the cell type - \n",
    "#genotype interaction effect, relative to the allele that is coded as 2 in the dosage file.\n",
    "\n",
    "results_df = pd.read_csv(f'{WRKDIR}/results_biowulf/{cohort}.deconResults.nov2020.chr1.fdr.csv', sep='\\t', index_col=0)\n",
    "print('chrom1: '+str(results_df.shape))\n",
    "\n",
    "for chrom in CHROMOSOMES_sub:\n",
    "    next_df = pd.read_csv(f'{WRKDIR}/results_biowulf/{cohort}.deconResults.nov2020.chr{chrom}.fdr.csv', sep='\\t', index_col=0)\n",
    "    results_df = pd.concat([results_df,next_df])\n",
    "    print(f'plus chrom{chrom}: '+str(results_df.shape))\n",
    "\n",
    "results_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total genes covered: 16663\n"
     ]
    }
   ],
   "source": [
    "# all_genes_decon = list(set(results_df['GENEID'].tolist()))\n",
    "\n",
    "# genelist_to_file(all_genes_decon, 'deconeqtl.all.genes')\n",
    "\n",
    "all_genes_decon = [line.rstrip('\\n') for line in open(f'{WRKDIR}/results/deconeqtl.all.genes')]\n",
    "    \n",
    "print('total genes covered: '+str(len(all_genes_decon)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Significant genes:\n",
      "Microglia: 0\n",
      "Astrocyte: 0\n",
      "OPC: 34\n",
      "Oligodendrocyte: 741\n",
      "Neuron: 39\n",
      "\n",
      "Signficant variants:\n",
      "Microglia: 0\n",
      "Astrocyte: 0\n",
      "OPC: 1618\n",
      "Oligodendrocyte: 17493\n",
      "Neuron: 15817\n"
     ]
    }
   ],
   "source": [
    "##### find variants found to be signif in each cell type\n",
    "\n",
    "\n",
    " \n",
    "# cell1_sig_df = find_signif(cell1)\n",
    "# cell1_sig_vars,cell1_sig_genes = to_list(cell1_sig_df)    \n",
    "    \n",
    "# cell2_sig_df = find_signif(cell2)\n",
    "# cell2_sig_vars,cell2_sig_genes = to_list(cell2_sig_df)\n",
    "\n",
    "# cell3_sig_df = find_signif(cell3)\n",
    "# cell3_sig_vars,cell3_sig_genes = to_list(cell3_sig_df)\n",
    "\n",
    "# cell4_sig_df = find_signif(cell4)\n",
    "# cell4_sig_vars,cell4_sig_genes = to_list(cell4_sig_df)\n",
    "\n",
    "# cell5_sig_df = find_signif(cell5)\n",
    "# cell5_sig_vars,cell5_sig_genes = to_list(cell5_sig_df)\n",
    "\n",
    "\n",
    "# ## saving lists to files\n",
    "# genelist_to_file(cell1_sig_genes, f'deconeqtl.significant.fdr.genes.{cell1}')\n",
    "# genelist_to_file(cell2_sig_genes, f'deconeqtl.significant.fdr.genes.{cell2}')\n",
    "# genelist_to_file(cell3_sig_genes, f'deconeqtl.significant.fdr.genes.{cell3}')\n",
    "# genelist_to_file(cell4_sig_genes, f'deconeqtl.significant.fdr.genes.{cell4}')\n",
    "# genelist_to_file(cell5_sig_genes, f'deconeqtl.significant.fdr.genes.{cell5}')\n",
    "\n",
    "# genelist_to_file(cell1_sig_vars, f'deconeqtl.significant.fdr.variants.{cell1}')\n",
    "# genelist_to_file(cell2_sig_vars, f'deconeqtl.significant.fdr.variants.{cell2}')\n",
    "# genelist_to_file(cell3_sig_vars, f'deconeqtl.significant.fdr.variants.{cell3}')\n",
    "# genelist_to_file(cell4_sig_vars, f'deconeqtl.significant.fdr.variants.{cell4}')\n",
    "# genelist_to_file(cell5_sig_vars, f'deconeqtl.significant.fdr.variants.{cell5}')\n",
    "\n",
    "\n",
    "##loading in lists\n",
    "print('Significant genes:')\n",
    "cell1_sig_genes = file_to_list(f'deconeqtl.significant.fdr.genes.{cell1}', cell1)\n",
    "cell2_sig_genes = file_to_list(f'deconeqtl.significant.fdr.genes.{cell2}', cell2)\n",
    "cell3_sig_genes = file_to_list(f'deconeqtl.significant.fdr.genes.{cell3}', cell3)\n",
    "cell4_sig_genes = file_to_list(f'deconeqtl.significant.fdr.genes.{cell4}', cell4)\n",
    "cell5_sig_genes = file_to_list(f'deconeqtl.significant.fdr.genes.{cell5}', cell5)\n",
    "\n",
    "\n",
    "print('')\n",
    "print('Signficant variants:')\n",
    "cell1_sig_vars = file_to_list(f'deconeqtl.significant.fdr.variants.{cell1}',cell1)\n",
    "cell2_sig_vars = file_to_list(f'deconeqtl.significant.fdr.variants.{cell2}',cell2)\n",
    "cell3_sig_vars = file_to_list(f'deconeqtl.significant.fdr.variants.{cell3}',cell3)\n",
    "cell4_sig_vars = file_to_list(f'deconeqtl.significant.fdr.variants.{cell4}',cell4)\n",
    "cell5_sig_vars = file_to_list(f'deconeqtl.significant.fdr.variants.{cell5}',cell5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Nonsignificant genes:\n",
      "Microglia: 16663\n",
      "Astrocyte: 16663\n",
      "OPC: 16629\n",
      "Oligodendrocyte: 15922\n",
      "Neuron: 16624\n",
      "\n",
      "Nonsignficant variants:\n",
      "Microglia: 72241266\n",
      "Astrocyte: 72241266\n",
      "OPC: 72239648\n",
      "Oligodendrocyte: 72223773\n",
      "Neuron: 72225449\n"
     ]
    }
   ],
   "source": [
    "##### find variants found to be nonsignif in each cell type\n",
    "\n",
    "def find_nonsignif_vars(cell_type):\n",
    "    nonsig_df = results_df.loc[results_df[f'{cell_type}_bh_fdr'] > 0.05]\n",
    "    nonsig_vars = nonsig_df.index.tolist()\n",
    "    print(f'{cell_type} variants: '+str(len(nonsig_vars)))\n",
    "    return nonsig_vars\n",
    "\n",
    "def find_nonsignif_genes(cell_type_sig_genes, cell_type):\n",
    "    nonsig_genes = set(all_genes_decon) - set(cell_type_sig_genes)\n",
    "    print(f'{cell_type} genes: '+str(len(nonsig_genes)))\n",
    "    return nonsig_genes\n",
    "\n",
    "    \n",
    "    \n",
    "# cell1_nonsig_vars = find_nonsignif_vars(cell1)\n",
    "# cell1_nonsig_genes = find_nonsignif_genes(cell1_sig_genes,cell1)    \n",
    "    \n",
    "# cell2_nonsig_vars = find_nonsignif_vars(cell2)\n",
    "# cell2_nonsig_genes = find_nonsignif_genes(cell2_sig_genes,cell2) \n",
    "\n",
    "# cell3_nonsig_vars = find_nonsignif_vars(cell3)\n",
    "# cell3_nonsig_genes = find_nonsignif_genes(cell3_sig_genes,cell3) \n",
    "\n",
    "# cell4_nonsig_vars = find_nonsignif_vars(cell4)\n",
    "# cell4_nonsig_genes = find_nonsignif_genes(cell4_sig_genes,cell4) \n",
    "\n",
    "# cell5_nonsig_vars = find_nonsignif_vars(cell5)\n",
    "# cell5_nonsig_genes = find_nonsignif_genes(cell5_sig_genes,cell5) \n",
    "\n",
    "# # saving lists to files\n",
    "# genelist_to_file(cell1_nonsig_genes, f'deconeqtl.nonsignificant.fdr.genes.{cell1}')\n",
    "# genelist_to_file(cell2_nonsig_genes, f'deconeqtl.nonsignificant.fdr.genes.{cell2}')\n",
    "# genelist_to_file(cell3_nonsig_genes, f'deconeqtl.nonsignificant.fdr.genes.{cell3}')\n",
    "# genelist_to_file(cell4_nonsig_genes, f'deconeqtl.nonsignificant.fdr.genes.{cell4}')\n",
    "# genelist_to_file(cell5_nonsig_genes, f'deconeqtl.nonsignificant.fdr.genes.{cell5}')\n",
    "\n",
    "# genelist_to_file(cell1_nonsig_vars, f'deconeqtl.nonsignificant.fdr.variants.{cell1}')\n",
    "# genelist_to_file(cell2_nonsig_vars, f'deconeqtl.nonsignificant.fdr.variants.{cell2}')\n",
    "# genelist_to_file(cell3_nonsig_vars, f'deconeqtl.nonsignificant.fdr.variants.{cell3}')\n",
    "# genelist_to_file(cell4_nonsig_vars, f'deconeqtl.nonsignificant.fdr.variants.{cell4}')\n",
    "# genelist_to_file(cell5_nonsig_vars, f'deconeqtl.nonsignificant.fdr.variants.{cell5}')\n",
    "\n",
    "\n",
    "\n",
    "##loading in lists\n",
    "print('Nonsignificant genes:')\n",
    "cell1_nonsig_genes = file_to_list(f'deconeqtl.nonsignificant.fdr.genes.{cell1}', cell1)\n",
    "cell2_nonsig_genes = file_to_list(f'deconeqtl.nonsignificant.fdr.genes.{cell2}', cell2)\n",
    "cell3_nonsig_genes = file_to_list(f'deconeqtl.nonsignificant.fdr.genes.{cell3}', cell3)\n",
    "cell4_nonsig_genes = file_to_list(f'deconeqtl.nonsignificant.fdr.genes.{cell4}', cell4)\n",
    "cell5_nonsig_genes = file_to_list(f'deconeqtl.nonsignificant.fdr.genes.{cell5}', cell5)\n",
    "\n",
    "\n",
    "print('')\n",
    "print('Nonsignficant variants:')\n",
    "cell1_nonsig_vars = file_to_list(f'deconeqtl.nonsignificant.fdr.variants.{cell1}',cell1)\n",
    "cell2_nonsig_vars = file_to_list(f'deconeqtl.nonsignificant.fdr.variants.{cell2}',cell2)\n",
    "cell3_nonsig_vars = file_to_list(f'deconeqtl.nonsignificant.fdr.variants.{cell3}',cell3)\n",
    "cell4_nonsig_vars = file_to_list(f'deconeqtl.nonsignificant.fdr.variants.{cell4}',cell4)\n",
    "cell5_nonsig_vars = file_to_list(f'deconeqtl.nonsignificant.fdr.variants.{cell5}',cell5)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Load in plink bulk results to compare"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chrom1: (15995, 17)\n",
      "plus chrom2: (30228, 17)\n",
      "plus chrom3: (40394, 17)\n",
      "plus chrom4: (48403, 17)\n",
      "plus chrom5: (61537, 17)\n",
      "plus chrom6: (92701, 17)\n",
      "plus chrom7: (112683, 17)\n",
      "plus chrom8: (118803, 17)\n",
      "plus chrom9: (129890, 17)\n",
      "plus chrom10: (140077, 17)\n",
      "plus chrom11: (152663, 17)\n",
      "plus chrom12: (163495, 17)\n",
      "plus chrom13: (167440, 17)\n",
      "plus chrom14: (171244, 17)\n",
      "plus chrom15: (179757, 17)\n",
      "plus chrom16: (186377, 17)\n",
      "plus chrom17: (230093, 17)\n",
      "plus chrom18: (232536, 17)\n",
      "plus chrom19: (249923, 17)\n",
      "plus chrom20: (253927, 17)\n",
      "plus chrom21: (258081, 17)\n",
      "plus chrom22: (263144, 17)\n"
     ]
    }
   ],
   "source": [
    "# ##load significant variants\n",
    "plink_df = pd.read_hdf(f'/labshare/raph/eqtl/{cohort}/plink/output/glm/{cohort}.aug2020.chr1.glm.hdf5')\n",
    "plink_df.head()\n",
    "plink_df['#CHROM'] = plink_df['#CHROM'].astype(str)\n",
    "plink_df['POS'] = plink_df['POS'].astype(str)\n",
    "plink_df['REF'] = plink_df['REF'].astype(str)\n",
    "plink_df['ALT'] = plink_df['ALT'].astype(str)\n",
    "plink_df['SNP'] = 'chr'+plink_df['#CHROM']+':'+plink_df['POS']+':'+plink_df['REF']+':'+plink_df['ALT']\n",
    "plink_df['PAIR'] = plink_df['gene_id']+'_'+plink_df['SNP']\n",
    "plink_sig = plink_df.loc[plink_df['bh_fdr'] < 0.05]\n",
    "print('chrom1: '+str(plink_sig.shape))\n",
    "\n",
    "for chrom in CHROMOSOMES_sub:\n",
    "    next_df = pd.read_hdf(f'/labshare/raph/eqtl/{cohort}/plink/output/glm/{cohort}.aug2020.chr{chrom}.glm.hdf5')\n",
    "    \n",
    "    next_df['#CHROM'] = next_df['#CHROM'].astype(str)\n",
    "    next_df['POS'] = next_df['POS'].astype(str)\n",
    "    next_df['REF'] = next_df['REF'].astype(str)\n",
    "    next_df['ALT'] = next_df['ALT'].astype(str)\n",
    "    next_df['SNP'] = 'chr'+next_df['#CHROM']+':'+next_df['POS']+':'+next_df['REF']+':'+next_df['ALT']\n",
    "    next_df['PAIR'] = next_df['gene_id']+'_'+next_df['SNP']\n",
    "\n",
    "    ## non signif bulk pairs\n",
    "    next_sig = next_df.loc[next_df['bh_fdr'] < 0.05]\n",
    "\n",
    "    plink_sig = pd.concat([plink_sig,next_sig])\n",
    "    print(f'plus chrom{chrom}: '+str(plink_sig.shape))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9007\n",
      "263144\n"
     ]
    }
   ],
   "source": [
    "plink_sig_genes = list(set(plink_sig['gene_id'].tolist()))\n",
    "print(len(plink_sig_genes))\n",
    "plink_sig_vars = plink_sig['PAIR'].tolist()\n",
    "print(len(plink_sig_vars))\n",
    "\n",
    "# ## saving list to file\n",
    "genelist_to_file(plink_sig_genes, 'plink.eqtl.significant.genes')\n",
    "genelist_to_file(plink_sig_vars, 'plink.eqtl.significant.variants')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chrom1: (9361342, 17)\n",
      "plus chrom2: (16540548, 17)\n",
      "plus chrom3: (22848540, 17)\n",
      "plus chrom4: (27508649, 17)\n",
      "plus chrom5: (32769992, 17)\n",
      "plus chrom6: (39973600, 17)\n",
      "plus chrom7: (45447560, 17)\n",
      "plus chrom8: (49667451, 17)\n",
      "plus chrom9: (53744946, 17)\n",
      "plus chrom10: (57927128, 17)\n",
      "plus chrom11: (63404111, 17)\n",
      "plus chrom12: (69352218, 17)\n",
      "plus chrom13: (71656824, 17)\n",
      "plus chrom14: (75390984, 17)\n",
      "plus chrom15: (78900649, 17)\n",
      "plus chrom16: (83501800, 17)\n",
      "plus chrom17: (89168326, 17)\n",
      "plus chrom18: (91116993, 17)\n",
      "plus chrom19: (98501090, 17)\n",
      "plus chrom20: (101086524, 17)\n",
      "plus chrom21: (102260890, 17)\n",
      "plus chrom22: (104712693, 17)\n"
     ]
    }
   ],
   "source": [
    "# ##load nonsignificant variants\n",
    "plink_df = pd.read_hdf(f'/labshare/raph/eqtl/nabec/plink/output/glm/{cohort}.aug2020.chr1.glm.hdf5')\n",
    "plink_df.head()\n",
    "plink_df['#CHROM'] = plink_df['#CHROM'].astype(str)\n",
    "plink_df['POS'] = plink_df['POS'].astype(str)\n",
    "plink_df['REF'] = plink_df['REF'].astype(str)\n",
    "plink_df['ALT'] = plink_df['ALT'].astype(str)\n",
    "plink_df['SNP'] = 'chr'+plink_df['#CHROM']+':'+plink_df['POS']+':'+plink_df['REF']+':'+plink_df['ALT']\n",
    "plink_df['PAIR'] = plink_df['gene_id']+'_'+plink_df['SNP']\n",
    "plink_nonsig = plink_df.loc[plink_df['bh_fdr'] > 0.05]\n",
    "print('chrom1: '+str(plink_nonsig.shape))\n",
    "\n",
    "for chrom in CHROMOSOMES_sub:\n",
    "    next_df = pd.read_hdf(f'/labshare/raph/eqtl/nabec/plink/output/glm/{cohort}.aug2020.chr{chrom}.glm.hdf5')\n",
    "    \n",
    "    next_df['#CHROM'] = next_df['#CHROM'].astype(str)\n",
    "    next_df['POS'] = next_df['POS'].astype(str)\n",
    "    next_df['REF'] = next_df['REF'].astype(str)\n",
    "    next_df['ALT'] = next_df['ALT'].astype(str)\n",
    "    next_df['SNP'] = 'chr'+next_df['#CHROM']+':'+next_df['POS']+':'+next_df['REF']+':'+next_df['ALT']\n",
    "    next_df['PAIR'] = next_df['gene_id']+'_'+next_df['SNP']\n",
    "\n",
    "    ## non signif bulk pairs\n",
    "    next_nonsig = next_df.loc[next_df['bh_fdr'] > 0.05]\n",
    "\n",
    "    plink_nonsig = pd.concat([plink_nonsig,next_nonsig])\n",
    "    print(f'plus chrom{chrom}: '+str(plink_nonsig.shape))\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16670\n",
      "104712693\n"
     ]
    }
   ],
   "source": [
    "plink_nonsig_genes = list(set(plink_nonsig['gene_id'].tolist()))\n",
    "print(len(plink_nonsig_genes))\n",
    "plink_nonsig_vars = plink_nonsig['PAIR'].tolist()\n",
    "print(len(plink_nonsig_vars))\n",
    "\n",
    "# ## saving list to file\n",
    "genelist_to_file(plink_nonsig_genes, 'plink.eqtl.nonsignificant.genes')\n",
    "genelist_to_file(plink_nonsig_vars, 'plink.eqtl.nonsignificant.variants')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16670\n",
      "104975837\n"
     ]
    }
   ],
   "source": [
    "##all\n",
    "print(len(list(set(plink_sig_genes + plink_nonsig_genes))))\n",
    "print(len(plink_sig_vars + plink_nonsig_vars))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# #plink_df = pd.read_csv(f'/labshare/anni/eqtl/ppmi/plink/output/eqtl/V08.genes.chr1.plink.glm.linear.gz', sep = '\\t')\n",
    "# plink_df = pd.read_csv(f'/labshare/raph/eqtl/nabec/plink/output/glm/{cohort}.aug2020.chr1.glm.hdf5', sep = '\\t')\n",
    "\n",
    "\n",
    "# #plink_df['PAIR'] = plink_df['GENEID']+'_'+plink_df['ID']\n",
    "# #plink_nonsig = plink_df.loc[plink_df['P'] > 0.05]\n",
    "# print('chrom1: '+str(plink_df.shape))\n",
    "\n",
    "# for chrom in CHROMOSOMES_sub:\n",
    "#     #next_df = pd.read_csv(f'/labshare/anni/eqtl/ppmi/plink/output/eqtl/V08.genes.chr{chrom}.plink.glm.linear.gz', sep = '\\t')\n",
    "#     next_df = pd.read_csv(f'/labshare/anni/eqtl/nabec/plink/output/glm/{cohort}.aug2020.chr{chrom}.glm.hdf5', sep = '\\t')\n",
    "\n",
    "\n",
    "#     #next_df['PAIR'] = next_df['GENEID']+'_'+next_df['ID']\n",
    "\n",
    "#     ## non signif bulk pairs\n",
    "#     #next_nonsig = next_df.loc[next_df['P'] > 0.05]\n",
    "\n",
    "#     plink_df = pd.concat([plink_df,next_df])\n",
    "#     print(f'plus chrom{chrom}: '+str(plink_df.shape))\n",
    "\n",
    "# plink_genes = list(set(plink_nonsig['GENEID'].tolist()))\n",
    "# plink_vars = list(set(plink_nonsig['PAIR'].tolist()))\n",
    "# plink_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# ##load significant variants\n",
    "# #plink_df = pd.read_csv(f'/labshare/anni/eqtl/ppmi/plink/output/eqtl/V08.genes.chr1.plink.glm.linear.gz', sep = '\\t')\n",
    "# plink_sig = pd.read_csv(f'/labshare/anni/eqtl/ppmi/plink/output/eqtl/V08.chr1.suggestive.plink.glm.linear.gz', sep = '\\t')\n",
    "\n",
    "\n",
    "# plink_sig['PAIR'] = plink_sig['GENEID']+'_'+plink_sig['ID']\n",
    "# #print(plink_df.shape)\n",
    "# plink_sig = plink_sig.loc[plink_sig['P'] < 0.05]\n",
    "# #print(plink_sig.shape)\n",
    "# print('chrom1: '+str(plink_sig.shape))\n",
    "\n",
    "# for chrom in CHROMOSOMES_sub:\n",
    "#     #next_df = pd.read_csv(f'/labshare/anni/eqtl/ppmi/plink/output/eqtl/V08.genes.chr{chrom}.plink.glm.linear.gz', sep = '\\t')\n",
    "#     next_df = pd.read_csv(f'/labshare/anni/eqtl/ppmi/plink/output/eqtl/V08.chr{chrom}.suggestive.plink.glm.linear.gz', sep = '\\t')\n",
    "#     next_df['PAIR'] = next_df['GENEID']+'_'+next_df['ID']\n",
    "\n",
    "#     ## non signif bulk pairs\n",
    "#     next_df = next_df.loc[next_df['P'] < 0.05]\n",
    "\n",
    "#     plink_sig = pd.concat([plink_sig,next_df])\n",
    "#     print(f'plus chrom{chrom}: '+str(plink_sig.shape))\n",
    "    \n",
    "# plink_sig_genes = list(set(plink_sig['GENEID'].tolist()))\n",
    "# plink_sig_vars = list(set(plink_sig['PAIR'].tolist()))\n",
    "\n",
    "# plink_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "All: 9007\n",
      "All: 263144\n",
      "All: 104712693\n"
     ]
    }
   ],
   "source": [
    "\n",
    "\n",
    "##loading in lists from file\n",
    "plink_sig_genes = file_to_list('plink.eqtl.significant.genes','All')\n",
    "plink_sig_vars = file_to_list('plink.eqtl.significant.variants','All')\n",
    "\n",
    "\n",
    "# saving list to file\n",
    "#genelist_to_file(plink_nonsig_genes, 'plink.eqtl.nonsignificant.genes')\n",
    "#genelist_to_file(plink_nonsig_vars, 'plink.eqtl.nonsignificant.variants')\n",
    "\n",
    "#loading in lists from file\n",
    "#plink_nonsig_genes = file_to_list('plink.eqtl.nonsignificant.genes')\n",
    "plink_nonsig_vars = file_to_list('plink.eqtl.nonsignificant.variants','All')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "9007"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test = list(set(plink_nonsig_genes) & set(plink_sig_genes))\n",
    "len(test)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2201\n",
      "199\n",
      "2002\n"
     ]
    }
   ],
   "source": [
    "# print(len(_sig_genes))\n",
    "# baso_noplink = set(baso_sig_genes) - set(plink_sig_genes)\n",
    "# print(len(baso_noplink))\n",
    "# baso_plink = set(baso_sig_genes) & set(plink_sig_genes)\n",
    "# print(len(baso_plink))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Significant in cell type of interest, not in plink bulk"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Microglia: 0\n",
      "Astrocyte: 0\n",
      "OPC: 17\n",
      "Oligodendrocyte: 252\n",
      "Neuron: 0\n",
      "variants\n",
      "Microglia: 0\n",
      "Astrocyte: 0\n",
      "OPC: 1558\n",
      "Oligodendrocyte: 17485\n",
      "Neuron: 0\n"
     ]
    }
   ],
   "source": [
    "\n",
    "outdir = '/labshare/anni/eqtl/Decon-eQTL/ppmi/results'\n",
    "\n",
    "# def nobulk_cell_genes(cell):\n",
    "#     cell_sig_genes = f'{cell}_sig_genes'\n",
    "#     nobulk_cell_genes = compare_plinknonsig(cell_sig_genes, cell)\n",
    "#     genelist_to_file(nobulk_cell_genes, f'decon_sig_plink_nonsig/{cohort}.{cell}_sig.fdr.plink_nonsig.genes')\n",
    "    \n",
    "# def nobulk_cell_variants(cell):\n",
    "#     cell_sig_vars = f'{cell}_sig_vars'\n",
    "#     nobulk_cell_vars = compare(plink_nonsig_vars, cell_sig_vars, cell)\n",
    "#     genelist_to_file(nobulk_cell_vars, f'decon_sig_plink_nonsig/{cohort}.{cell}_sig.fdr.plink_nonsig.variants')\n",
    "    \n",
    "\n",
    "    \n",
    "# print('genes signif in cell type, not in bulk')\n",
    "# for x in cell_types:\n",
    "#     nobulk_cell_genes(x)\n",
    "#     nobulk_cell_variants(x)\n",
    "\n",
    "nobulk_cell1_genes = compare_plinknonsig(cell1_sig_genes, cell1)\n",
    "nobulk_cell2_genes = compare_plinknonsig(cell2_sig_genes, cell2)\n",
    "nobulk_cell3_genes = compare_plinknonsig(cell3_sig_genes, cell3)\n",
    "nobulk_cell4_genes = compare_plinknonsig(cell4_sig_genes, cell4)\n",
    "nobulk_cell5_genes = compare_plinknonsig(cell5_sig_genes, cell5)\n",
    "\n",
    "## saving gene lists to files           \n",
    "genelist_to_file(nobulk_cell1_genes, f'decon_sig_plink_nonsig/{cohort}.{cell1}_sig.fdr.plink_nonsig.genes')\n",
    "genelist_to_file(nobulk_cell2_genes, f'decon_sig_plink_nonsig/{cohort}.{cell2}_sig.fdr.plink_nonsig.genes')\n",
    "genelist_to_file(nobulk_cell3_genes, f'decon_sig_plink_nonsig/{cohort}.{cell3}_sig.fdr.plink_nonsig.genes')\n",
    "genelist_to_file(nobulk_cell4_genes, f'decon_sig_plink_nonsig/{cohort}.{cell4}_sig.fdr.plink_nonsig.genes')\n",
    "genelist_to_file(nobulk_cell5_genes, f'decon_sig_plink_nonsig/{cohort}.{cell5}_sig.fdr.plink_nonsig.genes')\n",
    "\n",
    "\n",
    "print('variants')\n",
    "nobulk_cell1_vars = compare(plink_nonsig_vars, cell1_sig_vars, cell1)\n",
    "nobulk_cell2_vars = compare(plink_nonsig_vars, cell2_sig_vars, cell2)\n",
    "nobulk_cell3_vars = compare(plink_nonsig_vars, cell3_sig_vars, cell3)\n",
    "nobulk_cell4_vars = compare(plink_nonsig_vars, cell4_sig_vars, cell4)\n",
    "nobulk_cell5_vars = compare(plink_nonsig_vars, cell5_sig_vars, cell5)\n",
    "\n",
    "\n",
    "\n",
    "# saving variant lists to files           \n",
    "genelist_to_file(nobulk_cell1_vars, f'decon_sig_plink_nonsig/{cohort}.{cell1}_sig.fdr.plink_nonsig.variants')\n",
    "genelist_to_file(nobulk_cell2_vars, f'decon_sig_plink_nonsig/{cohort}.{cell2}_sig.fdr.plink_nonsig.variants')\n",
    "genelist_to_file(nobulk_cell3_vars, f'decon_sig_plink_nonsig/{cohort}.{cell3}_sig.fdr.plink_nonsig.variants')\n",
    "genelist_to_file(nobulk_cell4_vars, f'decon_sig_plink_nonsig/{cohort}.{cell4}_sig.fdr.plink_nonsig.variants')\n",
    "genelist_to_file(nobulk_cell5_vars, f'decon_sig_plink_nonsig/{cohort}.{cell5}_sig.fdr.plink_nonsig.variants')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Significant in bulk not in cell type of interest"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "genes signif in bulk, not in cell type\n",
      "Microglia: 9003\n",
      "Astrocyte: 9003\n",
      "OPC: 8986\n",
      "Oligodendrocyte: 8514\n",
      "Neuron: 8964\n",
      "variants\n",
      "Microglia: 224474\n",
      "Astrocyte: 224474\n",
      "OPC: 224414\n",
      "Oligodendrocyte: 224466\n",
      "Neuron: 208657\n"
     ]
    }
   ],
   "source": [
    "## comparing significant genes in plink bulk, not in the cell types\n",
    "# find overlap\n",
    "\n",
    "# def bulk_nocell_genes(cell):\n",
    "#     cell_nonsig_genes = f'{cell}_nonsig_genes'\n",
    "#     bulk_nocell_genes = compare(plink_sig_genes, cell_nonsig_genes, cell)\n",
    "#     genelist_to_file(bulk_nocell_genes, f'decon_nonsig_plink_sig/{cohort}.{cell}_nonsig.fdr.plink_sig.genes')\n",
    "    \n",
    "# def bulk_nocell_variants(cell):\n",
    "#     cell_nonsig_vars = f'{cell}_nonsig_vars'\n",
    "#     bulk_nocell_vars = compare(plink_sig_genes, cell_nonsig_vars, cell)\n",
    "#     genelist_to_file(bulk_nocell_vars, f'decon_nonsig_plink_sig/{cohort}.{cell}_nonsig.fdr.plink_sig.variants')\n",
    "    \n",
    "   \n",
    "# print('genes signif in bulk, not in cell type')\n",
    "# for x in cell_types:\n",
    "#     bulk_nocell_genes(x)\n",
    "#     bulk_nocell_variants(x)\n",
    "    \n",
    "    \n",
    "    \n",
    "\n",
    "print('genes signif in bulk, not in cell type')\n",
    "bulk_nocell1_genes = compare(plink_sig_genes, cell1_nonsig_genes, cell1)\n",
    "bulk_nocell2_genes = compare(plink_sig_genes, cell2_nonsig_genes, cell2)\n",
    "bulk_nocell3_genes = compare(plink_sig_genes, cell3_nonsig_genes, cell3)\n",
    "bulk_nocell4_genes = compare(plink_sig_genes, cell4_nonsig_genes, cell4)\n",
    "bulk_nocell5_genes = compare(plink_sig_genes, cell5_nonsig_genes, cell5)\n",
    "\n",
    "## saving gene lists to files\n",
    "genelist_to_file(bulk_nocell1_genes, f'decon_nonsig_plink_sig/{cohort}.{cell1}_nonsig.fdr.plink_sig.genes')\n",
    "genelist_to_file(bulk_nocell2_genes, f'decon_nonsig_plink_sig/{cohort}.{cell2}_nonsig.fdr.plink_sig.genes')\n",
    "genelist_to_file(bulk_nocell3_genes, f'decon_nonsig_plink_sig/{cohort}.{cell3}_nonsig.fdr.plink_sig.genes')\n",
    "genelist_to_file(bulk_nocell4_genes, f'decon_nonsig_plink_sig/{cohort}.{cell4}_nonsig.fdr.plink_sig.genes')\n",
    "genelist_to_file(bulk_nocell5_genes, f'decon_nonsig_plink_sig/{cohort}.{cell5}_nonsig.fdr.plink_sig.genes')\n",
    "\n",
    "\n",
    "print('variants')\n",
    "bulk_nocell1_vars = compare(plink_sig_vars, cell1_nonsig_vars, cell1)\n",
    "bulk_nocell2_vars = compare(plink_sig_vars, cell2_nonsig_vars, cell2)\n",
    "bulk_nocell3_vars = compare(plink_sig_vars, cell3_nonsig_vars, cell3)\n",
    "bulk_nocell4_vars = compare(plink_sig_vars, cell4_nonsig_vars, cell4)\n",
    "bulk_nocell5_vars = compare(plink_sig_vars, cell5_nonsig_vars, cell5)\n",
    "\n",
    "\n",
    "\n",
    "## saving gene lists to files           \n",
    "genelist_to_file(bulk_nocell1_vars, f'decon_nonsig_plink_sig/{cohort}.{cell1}_nonsig.fdr.plink_sig.variants')\n",
    "genelist_to_file(bulk_nocell2_vars, f'decon_nonsig_plink_sig/{cohort}.{cell2}_nonsig.fdr.plink_sig.variants')\n",
    "genelist_to_file(bulk_nocell3_vars, f'decon_nonsig_plink_sig/{cohort}.{cell3}_nonsig.fdr.plink_sig.variants')\n",
    "genelist_to_file(bulk_nocell4_vars, f'decon_nonsig_plink_sig/{cohort}.{cell4}_nonsig.fdr.plink_sig.variants')\n",
    "genelist_to_file(bulk_nocell5_vars, f'decon_nonsig_plink_sig/{cohort}.{cell5}_nonsig.fdr.plink_sig.variants')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Significant in both cell type of interest and in plink bulk"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "significant genes in bulk and decon:\n",
      "Microglia: 0\n",
      "Astrocyte: 0\n",
      "OPC: 17\n",
      "Oligodendrocyte: 489\n",
      "Neuron: 39\n",
      "significant variants in bulk and decon:\n",
      "Microglia: 0\n",
      "Astrocyte: 0\n",
      "OPC: 60\n",
      "Oligodendrocyte: 8\n",
      "Neuron: 15817\n"
     ]
    }
   ],
   "source": [
    "# # comparing significant genes in different cell types and overall bulk\n",
    "# # find overlap\n",
    "\n",
    "\n",
    "# def bulk_cell_genes(cell):\n",
    "#     cell_sig_genes = f'{cell}_sig_genes'\n",
    "#     bulk_cell_genes = compare(plink_sig_genes, cell_sig_genes, cell)\n",
    "#     genelist_to_file(bulk_cell_genes, f'decon_sig_plink_sig/{cohort}.{cell}_sig.fdr.plink_sig.genes')\n",
    "    \n",
    "# def bulk_cell_variants(cell):\n",
    "#     cell_sig_vars = f'{cell}_sig_vars'\n",
    "#     bulk_cell_vars = compare(plink_sig_genes, cell_sig_vars, cell)\n",
    "#     genelist_to_file(bulk_cell_vars, f'decon_sig_plink_sig/{cohort}.{cell}_sig.fdr.plink_sig.variants')\n",
    "    \n",
    "   \n",
    "# print('genes signif in bulk and cell type')\n",
    "# for x in cell_types:\n",
    "#     bulk_cell_genes(x)\n",
    "#     bulk_cell_variants(x)\n",
    "\n",
    "\n",
    "\n",
    "print('significant genes in bulk and decon:')\n",
    "bulk_cell1_genes = compare(plink_sig_genes, cell1_sig_genes, cell1)\n",
    "bulk_cell2_genes = compare(plink_sig_genes, cell2_sig_genes, cell2)\n",
    "bulk_cell3_genes = compare(plink_sig_genes, cell3_sig_genes, cell3)\n",
    "bulk_cell4_genes = compare(plink_sig_genes, cell4_sig_genes, cell4)\n",
    "bulk_cell5_genes = compare(plink_sig_genes, cell5_sig_genes, cell5)\n",
    "\n",
    "## saving gene lists to files\n",
    "genelist_to_file(bulk_cell1_genes, f'decon_sig_plink_sig/{cohort}.{cell1}_sig.fdr.plink_sig.genes')\n",
    "genelist_to_file(bulk_cell2_genes, f'decon_sig_plink_sig/{cohort}.{cell2}_sig.fdr.plink_sig.genes')\n",
    "genelist_to_file(bulk_cell3_genes, f'decon_sig_plink_sig/{cohort}.{cell3}_sig.fdr.plink_sig.genes')\n",
    "genelist_to_file(bulk_cell4_genes, f'decon_sig_plink_sig/{cohort}.{cell4}_sig.fdr.plink_sig.genes')\n",
    "genelist_to_file(bulk_cell5_genes, f'decon_sig_plink_sig/{cohort}.{cell5}_sig.fdr.plink_sig.genes')\n",
    "\n",
    "print('significant variants in bulk and decon:')\n",
    "bulk_cell1_vars = compare(plink_sig_vars, cell1_sig_vars, cell1)\n",
    "bulk_cell2_vars = compare(plink_sig_vars, cell2_sig_vars, cell2)\n",
    "bulk_cell3_vars = compare(plink_sig_vars, cell3_sig_vars, cell3)\n",
    "bulk_cell4_vars = compare(plink_sig_vars, cell4_sig_vars, cell4)\n",
    "bulk_cell5_vars = compare(plink_sig_vars, cell5_sig_vars, cell5)\n",
    "\n",
    "## saving gene lists to files\n",
    "genelist_to_file(bulk_cell1_vars, f'decon_sig_plink_sig/{cohort}.{cell1}_sig.fdr.plink_sig.variants')\n",
    "genelist_to_file(bulk_cell2_vars, f'decon_sig_plink_sig/{cohort}.{cell2}_sig.fdr.plink_sig.variants')\n",
    "genelist_to_file(bulk_cell3_vars, f'decon_sig_plink_sig/{cohort}.{cell3}_sig.fdr.plink_sig.variants')\n",
    "genelist_to_file(bulk_cell4_vars, f'decon_sig_plink_sig/{cohort}.{cell4}_sig.fdr.plink_sig.variants')\n",
    "genelist_to_file(bulk_cell5_vars, f'decon_sig_plink_sig/{cohort}.{cell5}_sig.fdr.plink_sig.variants')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## PUlling up specific gene lists"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "neutrophil: 44\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "['ENSG00000117115.13',\n",
       " 'ENSG00000165650.12',\n",
       " 'ENSG00000106772.18',\n",
       " 'ENSG00000163993.7',\n",
       " 'ENSG00000257433.6',\n",
       " 'ENSG00000277879.1',\n",
       " 'ENSG00000162069.16',\n",
       " 'ENSG00000123838.11',\n",
       " 'ENSG00000260286.4',\n",
       " 'ENSG00000091039.17',\n",
       " 'ENSG00000279996.1',\n",
       " 'ENSG00000005020.13',\n",
       " 'ENSG00000170448.12',\n",
       " 'ENSG00000256660.6',\n",
       " 'ENSG00000241764.4',\n",
       " 'ENSG00000135953.11',\n",
       " 'ENSG00000258476.6',\n",
       " 'ENSG00000172322.14',\n",
       " 'ENSG00000148110.16',\n",
       " 'ENSG00000127946.17',\n",
       " 'ENSG00000256751.6',\n",
       " 'ENSG00000163293.12',\n",
       " 'ENSG00000171617.14',\n",
       " 'ENSG00000260943.1',\n",
       " 'ENSG00000278745.1',\n",
       " 'ENSG00000072401.15',\n",
       " 'ENSG00000124588.20',\n",
       " 'ENSG00000129295.9',\n",
       " 'ENSG00000188056.12',\n",
       " 'ENSG00000112299.8',\n",
       " 'ENSG00000259235.2',\n",
       " 'ENSG00000287733.1',\n",
       " 'ENSG00000110203.9',\n",
       " 'ENSG00000285106.2',\n",
       " 'ENSG00000116299.17',\n",
       " 'ENSG00000163803.13',\n",
       " 'ENSG00000234426.3',\n",
       " 'ENSG00000184106.8',\n",
       " 'ENSG00000164251.5',\n",
       " 'ENSG00000286830.1',\n",
       " 'ENSG00000286480.1',\n",
       " 'ENSG00000242110.8',\n",
       " 'ENSG00000274256.1',\n",
       " 'ENSG00000237803.6']"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "neutro_sig_noplink = file_to_list('decon_sig_plink_nonsig/ppmi.neutrophil_sig.fdr.plink_nonsig.genes', 'neutrophil')\n",
    "neutro_sig_noplink"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
