{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Notebook for seting up resources to quantifying RNA with Salmon on GCP\n",
    "\n",
    "https://salmon.readthedocs.io/en/latest/salmon.html\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Jan 07, 2020"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [],
   "source": [
    "# setup notebook global vars\n",
    "#gcp\n",
    "project_id = 'nih-nia-lng-cbg'\n",
    "prj_bucket = 'gs://nihnialngcbg-eqtl'\n",
    "my_user='gibbsr'\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### pull Gencode reference transcripts for indexing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [],
   "source": [
    "# pull Gencode reference transcripts for indexing\n",
    "#setup the local reference files on labseq for gencode_37 from GRCh38/hg38\n",
    "gencode_lastest = '32'\n",
    "gencode_src_path = f'ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human\\\n",
    "/release_{gencode_lastest}'\n",
    "gencode_gtf = f'gencode.v{gencode_lastest}.annotation.gtf.gz'\n",
    "gencode_fa = f'gencode.v{gencode_lastest}.transcripts.fa.gz'\n",
    "gencode_local_dir = f'/labseq/Genomes/GRCh38_hg38/gencode_{gencode_lastest}'\n",
    "\n",
    "#make sure local path is created\n",
    "!mkdir -p {gencode_local_dir}\n",
    "\n",
    "#pull the gencode reference\n",
    "!curl --silent -L {gencode_src_path}/{gencode_gtf} \\\n",
    "--output {gencode_local_dir}/{gencode_gtf}\n",
    "\n",
    "!curl --silent -L {gencode_src_path}/{gencode_fa} \\\n",
    "--output {gencode_local_dir}/{gencode_fa}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### prep reference transcriptome as need"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [],
   "source": [
    "# for Gencode\n",
    "# exclude transcripts with low level support, ie only keep 'goodish' \\\n",
    "# exclude Ensembl TSL levels 4 or 5 and Gencode level 3\n",
    "\n",
    "# need to use salmon decoy prep script (?)\n",
    "\n",
    "!gunzip {gencode_local_dir}/{gencode_fa}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [],
   "source": [
    "# for CHESS\n",
    "# need to create transcript fasta from GTF and reference genome\n",
    "# need to modify fasta entries labels with transcript and gene IDs and gene name\n",
    "\n",
    "# need to use salmon decoy prep script (?)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gsutil -mq cp /labseq/Genomes/GRCh38_hg38/gencode_32/gencode.v32.transcripts.fa gs://nihnialngcbg-eqtl/resources/references/gencode.v32.transcripts.fa\n"
     ]
    }
   ],
   "source": [
    "#### push prepped transcriptome(s) to cloud\n",
    "local_transcriptome_fasta = f'{gencode_local_dir}/{gencode_fa}'.replace('.gz','')\n",
    "gcs_transcriptome_fasta = f'{prj_bucket}/resources/references/{gencode_fa}'.replace('.gz','')\n",
    "\n",
    "gcloud_cmd = f'gsutil -mq cp {local_transcriptome_fasta} {gcs_transcriptome_fasta}'\n",
    "\n",
    "print(gcloud_cmd)\n",
    "\n",
    "!{gcloud_cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### create salmon index for the reference transcriptome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gcloud beta lifesciences pipelines run --project nih-nia-lng-cbg --docker-image us.gcr.io/nih-nia-lng-cbg/nihnialngcbg-salmon:2020-1 --machine-type n1-standard-4 --logging gs://nihnialngcbg-eqtl/resources/references/logs/salmon_index_gencode_32.log --command-line 'mkdir -p /gcloud-shared/reference; salmon --no-version-check index -t ${REF_TX_FASTA} -i /gcloud-shared/reference -k 31 --gencode --threads $(nproc); gsutil -mq cp -r /gcloud-shared/reference ${GS_SALMON_INDEX_OUT}; ls -lhR /gcloud-shared' --inputs REF_TX_FASTA=gs://nihnialngcbg-eqtl/resources/references/gencode.v32.transcripts.fa --env-vars GS_SALMON_INDEX_OUT=gs://nihnialngcbg-eqtl/resources/references/salmon/gencode_32 --labels=pipe=salmonindex,reference=gencode,user=gibbsr --preemptible\n",
      "\n",
      "Running [projects/463418702553/locations/us-central1/operations/7476798936617483009].\n"
     ]
    }
   ],
   "source": [
    "# use a GCP lifesciences job to generate the salmon index(?)\n",
    "docker_img = 'us.gcr.io/nih-nia-lng-cbg/nihnialngcbg-salmon:2020-1'\n",
    "\n",
    "# gcs_transcriptome_fasta = f'{prj_bucket}/resources/references/salmon/gencode_r32/\\\n",
    "# nohup.filtered.gencode.v32.transcripts.fa'\n",
    "\n",
    "gcp_lsci_pipe_cmd = 'mkdir -p /gcloud-shared/reference; \\\n",
    "salmon --no-version-check index -t ${REF_TX_FASTA} -i /gcloud-shared/reference \\\n",
    "-k 31 --gencode --threads $(nproc); \\\n",
    "gsutil -mq cp -r /gcloud-shared/reference ${GS_SALMON_INDEX_OUT}; \\\n",
    "ls -lhR /gcloud-shared'\n",
    "\n",
    "gcp_lsci_pipe_vars = f'GS_SALMON_INDEX_OUT={prj_bucket}/resources/references/\\\n",
    "salmon/gencode_{gencode_lastest}'\n",
    "\n",
    "cloud_cmd = f'gcloud beta lifesciences pipelines run \\\n",
    "--project {project_id} \\\n",
    "--docker-image {docker_img} \\\n",
    "--machine-type n1-standard-4 \\\n",
    "--logging {prj_bucket}/resources/references/logs/salmon_index_gencode_{gencode_lastest}.log \\\n",
    "--command-line \\'{gcp_lsci_pipe_cmd}\\' \\\n",
    "--inputs REF_TX_FASTA={gcs_transcriptome_fasta} \\\n",
    "--env-vars {gcp_lsci_pipe_vars} \\\n",
    "--labels=pipe=salmonindex,reference=gencode,user={my_user} \\\n",
    "--preemptible'\n",
    "\n",
    "print(f'{cloud_cmd}\\n')\n",
    "\n",
    "!{cloud_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "done: true\n",
      "metadata:\n",
      "  '@type': type.googleapis.com/google.cloud.lifesciences.v2beta.Metadata\n",
      "  createTime: '2020-01-09T21:39:49.740754Z'\n",
      "  endTime: '2020-01-09T21:52:40.707318167Z'\n",
      "  events:\n",
      "  - description: Worker released\n",
      "    timestamp: '2020-01-09T21:52:40.707318167Z'\n",
      "    workerReleased:\n",
      "      instance: google-pipelines-worker-b117596df4631df34df9ffb03442457f\n",
      "      zone: us-central1-f\n",
      "  - containerStopped:\n",
      "      actionId: 3\n",
      "    description: Stopped running \"/bin/sh -c gsutil -m -q cp /google/logs/output gs://nihnialngcbg-eqtl/resources/references/logs/salmon_index_gencode_32.log\"\n",
      "    timestamp: '2020-01-09T21:52:39.596493069Z'\n",
      "  - containerStarted:\n",
      "      actionId: 3\n",
      "    description: Started running \"/bin/sh -c gsutil -m -q cp /google/logs/output gs://nihnialngcbg-eqtl/resources/references/logs/salmon_index_gencode_32.log\"\n",
      "    timestamp: '2020-01-09T21:52:32.275574630Z'\n",
      "  - containerStopped:\n",
      "      actionId: 2\n",
      "      stderr: |\n",
      "[Building BooPHF]  99.8 %   elapsed:   0 min 14 sec   remaining:   0 min 0  sec\n",
      "        [2020-01-09 21:52:03.462] [puff::index::jointLog] [info] mphf size = 87.5192 MB\n",
      "        [2020-01-09 21:52:03.462] [puff::index::jointLog] [info] chunk size = 45706350\n",
      "        [2020-01-09 21:52:03.462] [puff::index::jointLog] [info] chunk 0 = [0, 45706350)\n",
      "        [2020-01-09 21:52:03.462] [puff::index::jointLog] [info] chunk 1 = [45706350, 91412700)\n",
      "        [2020-01-09 21:52:03.462] [puff::index::jointLog] [info] chunk 2 = [91412700, 137119072)\n",
      "        [2020-01-09 21:52:03.462] [puff::index::jointLog] [info] chunk 3 = [137119072, 182825367)\n",
      "        [2020-01-09 21:52:19.627] [puff::index::jointLog] [info] finished populating pos vector\n",
      "        [2020-01-09 21:52:19.627] [puff::index::jointLog] [info] writing index components\n",
      "        [2020-01-09 21:52:20.079] [puff::index::jointLog] [info] finished writing dense pufferfish index\n",
      "        [2020-01-09 21:52:20.189] [jLog] [info] done building index\n",
      "    description: |-\n",
      "[Building BooPHF]  99.8 %   elapsed:   0 min 14 sec   remaining:   0 min 0  secn-check index -t ${REF_TX_FASTA} -i /gcloud-shared/reference -k 31 --gencode --threads $(nproc); gsutil -mq cp -r /gcloud-shared/reference ${GS_SALMON_INDEX_OUT}; ls -lhR /gcloud-shared\": c   remaining:   0 min 0  sec\n",
      "      [2020-01-09 21:52:03.462] [puff::index::jointLog] [info] mphf size = 87.5192 MB\n",
      "      [2020-01-09 21:52:03.462] [puff::index::jointLog] [info] chunk size = 45706350\n",
      "      [2020-01-09 21:52:03.462] [puff::index::jointLog] [info] chunk 0 = [0, 45706350)\n",
      "      [2020-01-09 21:52:03.462] [puff::index::jointLog] [info] chunk 1 = [45706350, 91412700)\n",
      "      [2020-01-09 21:52:03.462] [puff::index::jointLog] [info] chunk 2 = [91412700, 137119072)\n",
      "      [2020-01-09 21:52:03.462] [puff::index::jointLog] [info] chunk 3 = [137119072, 182825367)\n",
      "      [2020-01-09 21:52:19.627] [puff::index::jointLog] [info] finished populating pos vector\n",
      "      [2020-01-09 21:52:19.627] [puff::index::jointLog] [info] writing index components\n",
      "      [2020-01-09 21:52:20.079] [puff::index::jointLog] [info] finished writing dense pufferfish index\n",
      "      [2020-01-09 21:52:20.189] [jLog] [info] done building index\n",
      "    timestamp: '2020-01-09T21:52:32.089901248Z'\n",
      "  - containerStarted:\n",
      "      actionId: 2\n",
      "    description: Started running \"-c mkdir -p /gcloud-shared/reference; salmon --no-version-check\n",
      "      index -t ${REF_TX_FASTA} -i /gcloud-shared/reference -k 31 --gencode --threads\n",
      "      $(nproc); gsutil -mq cp -r /gcloud-shared/reference ${GS_SALMON_INDEX_OUT};\n",
      "      ls -lhR /gcloud-shared\"\n",
      "    timestamp: '2020-01-09T21:41:25.431452825Z'\n",
      "  - containerStopped:\n",
      "      actionId: 1\n",
      "    description: Stopped running \"/bin/sh -c gsutil -m -q cp gs://nihnialngcbg-eqtl/resources/references/gencode.v32.transcripts.fa\n",
      "      ${REF_TX_FASTA}\"\n",
      "    timestamp: '2020-01-09T21:41:25.349287337Z'\n",
      "  - containerStarted:\n",
      "      actionId: 1\n",
      "    description: Started running \"/bin/sh -c gsutil -m -q cp gs://nihnialngcbg-eqtl/resources/references/gencode.v32.transcripts.fa\n",
      "      ${REF_TX_FASTA}\"\n",
      "    timestamp: '2020-01-09T21:41:19.004983823Z'\n",
      "  - description: Stopped pulling \"us.gcr.io/nih-nia-lng-cbg/nihnialngcbg-salmon:2020-1\"\n",
      "    pullStopped:\n",
      "      imageUri: us.gcr.io/nih-nia-lng-cbg/nihnialngcbg-salmon:2020-1\n",
      "    timestamp: '2020-01-09T21:41:09.914169180Z'\n",
      "  - description: Started pulling \"us.gcr.io/nih-nia-lng-cbg/nihnialngcbg-salmon:2020-1\"\n",
      "    pullStarted:\n",
      "      imageUri: us.gcr.io/nih-nia-lng-cbg/nihnialngcbg-salmon:2020-1\n",
      "    timestamp: '2020-01-09T21:40:35.988118855Z'\n",
      "  - description: Stopped pulling \"google/cloud-sdk:slim\"\n",
      "    pullStopped:\n",
      "      imageUri: google/cloud-sdk:slim\n",
      "    timestamp: '2020-01-09T21:40:35.950127379Z'\n",
      "  - description: Started pulling \"google/cloud-sdk:slim\"\n",
      "    pullStarted:\n",
      "      imageUri: google/cloud-sdk:slim\n",
      "    timestamp: '2020-01-09T21:40:21.389707480Z'\n",
      "  - description: Worker \"google-pipelines-worker-b117596df4631df34df9ffb03442457f\"\n",
      "      assigned in \"us-central1-f\"\n",
      "    timestamp: '2020-01-09T21:39:50.683864039Z'\n",
      "    workerAssigned:\n",
      "      instance: google-pipelines-worker-b117596df4631df34df9ffb03442457f\n",
      "      machineType: n1-standard-4\n",
      "      zone: us-central1-f\n",
      "  labels:\n",
      "    pipe: salmonindex\n",
      "    reference: gencode\n",
      "    user: gibbsr\n",
      "  pipeline:\n",
      "    actions:\n",
      "    - commands:\n",
      "      - /bin/sh\n",
      "      - -c\n",
      "      - gsutil -m -q cp gs://nihnialngcbg-eqtl/resources/references/gencode.v32.transcripts.fa\n",
      "        ${REF_TX_FASTA}\n",
      "      imageUri: google/cloud-sdk:slim\n",
      "      mounts:\n",
      "      - disk: gcloud-shared\n",
      "        path: /gcloud-shared\n",
      "    - commands:\n",
      "      - -c\n",
      "      - mkdir -p /gcloud-shared/reference; salmon --no-version-check index -t ${REF_TX_FASTA}\n",
      "        -i /gcloud-shared/reference -k 31 --gencode --threads $(nproc); gsutil -mq\n",
      "        cp -r /gcloud-shared/reference ${GS_SALMON_INDEX_OUT}; ls -lhR /gcloud-shared\n",
      "      entrypoint: bash\n",
      "      imageUri: us.gcr.io/nih-nia-lng-cbg/nihnialngcbg-salmon:2020-1\n",
      "      mounts:\n",
      "      - disk: gcloud-shared\n",
      "        path: /gcloud-shared\n",
      "    - alwaysRun: true\n",
      "      commands:\n",
      "      - /bin/sh\n",
      "      - -c\n",
      "      - gsutil -m -q cp /google/logs/output gs://nihnialngcbg-eqtl/resources/references/logs/salmon_index_gencode_32.log\n",
      "      imageUri: google/cloud-sdk:slim\n",
      "    environment:\n",
      "      GS_SALMON_INDEX_OUT: gs://nihnialngcbg-eqtl/resources/references/salmon/gencode_32\n",
      "      REF_TX_FASTA: /gcloud-shared/input0\n",
      "    resources:\n",
      "      virtualMachine:\n",
      "        bootDiskSizeGb: 10\n",
      "        bootImage: projects/cos-cloud/global/images/family/cos-stable\n",
      "        disks:\n",
      "        - name: gcloud-shared\n",
      "          sizeGb: 500\n",
      "        labels:\n",
      "          goog-pipelines-worker: 'true'\n",
      "        machineType: n1-standard-4\n",
      "        preemptible: true\n",
      "        serviceAccount:\n",
      "          email: default\n",
      "          scopes:\n",
      "          - https://www.googleapis.com/auth/cloud-platform\n",
      "      zones:\n",
      "      - us-central1-f\n",
      "    timeout: 604800s\n",
      "  startTime: '2020-01-09T21:39:50.683864039Z'\n",
      "name: projects/463418702553/locations/us-central1/operations/7476798936617483009\n",
      "response:\n",
      "  '@type': type.googleapis.com/cloud.lifesciences.pipelines.RunPipelineResponse\n"
     ]
    }
   ],
   "source": [
    "#check job status, id OPID above\n",
    "op_id = 7476798936617483009\n",
    "\n",
    "!gcloud beta lifesciences operations describe {op_id}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### run a test of quantifying an RNA sample"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gcloud beta lifesciences pipelines run --project nih-nia-lng-cbg --docker-image us.gcr.io/nih-nia-lng-cbg/nihnialngcbg-salmon:2020-1 --machine-type n1-standard-4 --logging gs://nihnialngcbg-eqtl/rosmap_test/logs/salmon/102_120418 --command-line 'mkdir -p /gcloud-shared/reference; gsutil -mq cp ${SALMONTRANSCRIPTMODELS}/* /gcloud-shared/reference; salmon --no-version-check quant -i /gcloud-shared/reference -l A -1 ${FASTQR1} -2 ${FASTQR2} --validateMappings --seqBias --gcBias --posBias --threads $(nproc) -o /gcloud-shared/sample; gsutil -mq cp -r /gcloud-shared/sample ${GS_OUTPATH}; ls -lhR /gcloud-shared' --inputs FASTQR1=gs://nihnialngcbg-eqtl/rosmap/fastqs/102_120418_R1.fastq.gz --inputs FASTQR2=gs://nihnialngcbg-eqtl/rosmap/fastqs/102_120418_R2.fastq.gz --env-vars SALMONTRANSCRIPTMODELS=gs://nihnialngcbg-eqtl/resources/references/salmon/gencode_32,GS_OUTPATH=gs://nihnialngcbg-eqtl/rosmap_test/quants/gencode_r32/102_120418,FASTQR1=/gcloud-shared/inputR1.gz,FASTQR2=/gcloud-shared/inputR2.gz --labels=pipe=salmonquant,sample=102_120418,user=gibbsr \n",
      "\n",
      "Running [projects/463418702553/locations/us-central1/operations/8359902396683357875].\n"
     ]
    }
   ],
   "source": [
    "# run a test of quantifying an RNA sample\n",
    "in_bucket = 'gs://nihnialngcbg-eqtl/rosmap'\n",
    "out_bucket = 'gs://nihnialngcbg-eqtl/rosmap_test'\n",
    "sample = '102_120418'\n",
    "\n",
    "gcp_lsci_pipe_cmd = 'mkdir -p /gcloud-shared/reference; \\\n",
    "gsutil -mq cp ${SALMONTRANSCRIPTMODELS}/* /gcloud-shared/reference; \\\n",
    "salmon --no-version-check quant -i /gcloud-shared/reference -l A -1 ${FASTQR1} -2 ${FASTQR2} \\\n",
    "--validateMappings --seqBias --gcBias --posBias --threads $(nproc) \\\n",
    "-o /gcloud-shared/sample; \\\n",
    "gsutil -mq cp -r /gcloud-shared/sample ${GS_OUTPATH}; \\\n",
    "ls -lhR /gcloud-shared'\n",
    "\n",
    "gcp_lsci_pipe_vars = f'SALMONTRANSCRIPTMODELS={prj_bucket}/resources/references/\\\n",
    "salmon/gencode_{gencode_lastest},\\\n",
    "GS_OUTPATH={out_bucket}/quants/gencode_r32/{sample},\\\n",
    "FASTQR1=/gcloud-shared/inputR1.gz,\\\n",
    "FASTQR2=/gcloud-shared/inputR2.gz'\n",
    "\n",
    "cloud_cmd = f'gcloud beta lifesciences pipelines run \\\n",
    "--project {project_id} \\\n",
    "--docker-image {docker_img} \\\n",
    "--machine-type n1-standard-4 \\\n",
    "--logging {out_bucket}/logs/salmon/{sample} \\\n",
    "--command-line \\'{gcp_lsci_pipe_cmd}\\' \\\n",
    "--inputs FASTQR1={in_bucket}/fastqs/{sample}_R1.fastq.gz \\\n",
    "--inputs FASTQR2={in_bucket}/fastqs/{sample}_R2.fastq.gz \\\n",
    "--env-vars {gcp_lsci_pipe_vars} \\\n",
    "--labels=pipe=salmonquant,sample={sample},user={my_user} '\n",
    "\n",
    "# ' \\\n",
    "# --preemptible'\n",
    "\n",
    "print(f'{cloud_cmd}\\n')\n",
    "\n",
    "!{cloud_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "done: true\n",
      "metadata:\n",
      "  '@type': type.googleapis.com/google.cloud.lifesciences.v2beta.Metadata\n",
      "  createTime: '2020-01-09T22:54:33.648924Z'\n",
      "  endTime: '2020-01-09T23:46:42.441497100Z'\n",
      "  events:\n",
      "  - description: Worker released\n",
      "    timestamp: '2020-01-09T23:46:42.441497100Z'\n",
      "    workerReleased:\n",
      "      instance: google-pipelines-worker-ef96c2fdee115da0ee33424069362bfa\n",
      "      zone: us-central1-f\n",
      "  - containerStopped:\n",
      "      actionId: 4\n",
      "    description: Stopped running \"/bin/sh -c gsutil -m -q cp /google/logs/output gs://nihnialngcbg-eqtl/rosmap_test/logs/salmon/102_120418\"\n",
      "    timestamp: '2020-01-09T23:46:41.624120594Z'\n",
      "  - containerStarted:\n",
      "      actionId: 4\n",
      "    description: Started running \"/bin/sh -c gsutil -m -q cp /google/logs/output gs://nihnialngcbg-eqtl/rosmap_test/logs/salmon/102_120418\"\n",
      "    timestamp: '2020-01-09T23:46:34.628396745Z'\n",
      "  - containerStopped:\n",
      "      actionId: 3\n",
      "      stderr: |+\n",
      "        rel diff. = 0.163479\n",
      "        [2020-01-09 23:43:26.470] [jointLog] [info] iteration = 700 | max rel diff. = 2.90496\n",
      "        [2020-01-09 23:43:49.135] [jointLog] [info] iteration = 800 | max rel diff. = 3.95903\n",
      "        [2020-01-09 23:44:11.879] [jointLog] [info] iteration = 900 | max rel diff. = 0.482702\n",
      "        [2020-01-09 23:44:34.759] [jointLog] [info] iteration = 1000 | max rel diff. = 0.0189542\n",
      "        [2020-01-09 23:44:57.706] [jointLog] [info] iteration = 1100 | max rel diff. = 0.13307\n",
      "        [2020-01-09 23:45:20.722] [jointLog] [info] iteration = 1200 | max rel diff. = 0.0340362\n",
      "        [2020-01-09 23:45:43.707] [jointLog] [info] iteration = 1300 | max rel diff. = 0.0243075\n",
      "        [2020-01-09 23:46:06.561] [jointLog] [info] iteration = 1400 | max rel diff. = 0.0346233\n",
      "        [2020-01-09 23:46:29.333] [jointLog] [info] iteration = 1500 | max rel diff. = 0.380716\n",
      "        [2020-01-09 23:46:30.484] [jointLog] [info] iteration = 1506 | max rel diff. = 0.0068032\n",
      "        [2020-01-09 23:46:30.495] [jointLog] [info] Finished optimizer\n",
      "        [2020-01-09 23:46:30.495] [jointLog] [info] writing output \n",
      "\n",
      "    description: |-\n",
      "      Stopped running \"-c mkdir -p /gcloud-shared/reference; gsutil -mq cp ${SALMONTRANSCRIPTMODELS}/* /gcloud-shared/reference; salmon --no-version-check quant -i /gcloud-shared/reference -l A -1 ${FASTQR1} -2 ${FASTQR2} --validateMappings --seqBias --gcBias --posBias --threads $(nproc) -o /gcloud-shared/sample; gsutil -mq cp -r /gcloud-shared/sample ${GS_OUTPATH}; ls -lhR /gcloud-shared\": rel diff. = 0.163479\n",
      "      [2020-01-09 23:43:26.470] [jointLog] [info] iteration = 700 | max rel diff. = 2.90496\n",
      "      [2020-01-09 23:43:49.135] [jointLog] [info] iteration = 800 | max rel diff. = 3.95903\n",
      "      [2020-01-09 23:44:11.879] [jointLog] [info] iteration = 900 | max rel diff. = 0.482702\n",
      "      [2020-01-09 23:44:34.759] [jointLog] [info] iteration = 1000 | max rel diff. = 0.0189542\n",
      "      [2020-01-09 23:44:57.706] [jointLog] [info] iteration = 1100 | max rel diff. = 0.13307\n",
      "      [2020-01-09 23:45:20.722] [jointLog] [info] iteration = 1200 | max rel diff. = 0.0340362\n",
      "      [2020-01-09 23:45:43.707] [jointLog] [info] iteration = 1300 | max rel diff. = 0.0243075\n",
      "      [2020-01-09 23:46:06.561] [jointLog] [info] iteration = 1400 | max rel diff. = 0.0346233\n",
      "      [2020-01-09 23:46:29.333] [jointLog] [info] iteration = 1500 | max rel diff. = 0.380716\n",
      "      [2020-01-09 23:46:30.484] [jointLog] [info] iteration = 1506 | max rel diff. = 0.0068032\n",
      "      [2020-01-09 23:46:30.495] [jointLog] [info] Finished optimizer\n",
      "      [2020-01-09 23:46:30.495] [jointLog] [info] writing output\n",
      "    timestamp: '2020-01-09T23:46:34.538898258Z'\n",
      "  - containerStarted:\n",
      "      actionId: 3\n",
      "    description: Started running \"-c mkdir -p /gcloud-shared/reference; gsutil -mq\n",
      "      cp ${SALMONTRANSCRIPTMODELS}/* /gcloud-shared/reference; salmon --no-version-check\n",
      "      quant -i /gcloud-shared/reference -l A -1 ${FASTQR1} -2 ${FASTQR2} --validateMappings\n",
      "      --seqBias --gcBias --posBias --threads $(nproc) -o /gcloud-shared/sample; gsutil\n",
      "      -mq cp -r /gcloud-shared/sample ${GS_OUTPATH}; ls -lhR /gcloud-shared\"\n",
      "    timestamp: '2020-01-09T22:57:02.049569878Z'\n",
      "  - containerStopped:\n",
      "      actionId: 2\n",
      "    description: Stopped running \"/bin/sh -c gsutil -m -q cp gs://nihnialngcbg-eqtl/rosmap/fastqs/102_120418_R1.fastq.gz\n",
      "      ${FASTQR1}\"\n",
      "    timestamp: '2020-01-09T22:57:01.968061290Z'\n",
      "  - containerStarted:\n",
      "      actionId: 2\n",
      "    description: Started running \"/bin/sh -c gsutil -m -q cp gs://nihnialngcbg-eqtl/rosmap/fastqs/102_120418_R1.fastq.gz\n",
      "      ${FASTQR1}\"\n",
      "    timestamp: '2020-01-09T22:56:41.730094455Z'\n",
      "  - containerStopped:\n",
      "      actionId: 1\n",
      "    description: Stopped running \"/bin/sh -c gsutil -m -q cp gs://nihnialngcbg-eqtl/rosmap/fastqs/102_120418_R2.fastq.gz\n",
      "      ${FASTQR2}\"\n",
      "    timestamp: '2020-01-09T22:56:41.602966844Z'\n",
      "  - containerStarted:\n",
      "      actionId: 1\n",
      "    description: Started running \"/bin/sh -c gsutil -m -q cp gs://nihnialngcbg-eqtl/rosmap/fastqs/102_120418_R2.fastq.gz\n",
      "      ${FASTQR2}\"\n",
      "    timestamp: '2020-01-09T22:56:02.348035277Z'\n",
      "  - description: Stopped pulling \"us.gcr.io/nih-nia-lng-cbg/nihnialngcbg-salmon:2020-1\"\n",
      "    pullStopped:\n",
      "      imageUri: us.gcr.io/nih-nia-lng-cbg/nihnialngcbg-salmon:2020-1\n",
      "    timestamp: '2020-01-09T22:55:54.660964210Z'\n",
      "  - description: Started pulling \"us.gcr.io/nih-nia-lng-cbg/nihnialngcbg-salmon:2020-1\"\n",
      "    pullStarted:\n",
      "      imageUri: us.gcr.io/nih-nia-lng-cbg/nihnialngcbg-salmon:2020-1\n",
      "    timestamp: '2020-01-09T22:55:19.817885912Z'\n",
      "  - description: Stopped pulling \"google/cloud-sdk:slim\"\n",
      "    pullStopped:\n",
      "      imageUri: google/cloud-sdk:slim\n",
      "    timestamp: '2020-01-09T22:55:19.769623863Z'\n",
      "  - description: Started pulling \"google/cloud-sdk:slim\"\n",
      "    pullStarted:\n",
      "      imageUri: google/cloud-sdk:slim\n",
      "    timestamp: '2020-01-09T22:55:05.170242327Z'\n",
      "  - description: Worker \"google-pipelines-worker-ef96c2fdee115da0ee33424069362bfa\"\n",
      "      assigned in \"us-central1-f\"\n",
      "    timestamp: '2020-01-09T22:54:34.541001915Z'\n",
      "    workerAssigned:\n",
      "      instance: google-pipelines-worker-ef96c2fdee115da0ee33424069362bfa\n",
      "      machineType: n1-standard-4\n",
      "      zone: us-central1-f\n",
      "  labels:\n",
      "    pipe: salmonquant\n",
      "    sample: '102_120418'\n",
      "    user: gibbsr\n",
      "  pipeline:\n",
      "    actions:\n",
      "    - commands:\n",
      "      - /bin/sh\n",
      "      - -c\n",
      "      - gsutil -m -q cp gs://nihnialngcbg-eqtl/rosmap/fastqs/102_120418_R2.fastq.gz\n",
      "        ${FASTQR2}\n",
      "      imageUri: google/cloud-sdk:slim\n",
      "      mounts:\n",
      "      - disk: gcloud-shared\n",
      "        path: /gcloud-shared\n",
      "    - commands:\n",
      "      - /bin/sh\n",
      "      - -c\n",
      "      - gsutil -m -q cp gs://nihnialngcbg-eqtl/rosmap/fastqs/102_120418_R1.fastq.gz\n",
      "        ${FASTQR1}\n",
      "      imageUri: google/cloud-sdk:slim\n",
      "      mounts:\n",
      "      - disk: gcloud-shared\n",
      "        path: /gcloud-shared\n",
      "    - commands:\n",
      "      - -c\n",
      "      - mkdir -p /gcloud-shared/reference; gsutil -mq cp ${SALMONTRANSCRIPTMODELS}/*\n",
      "        /gcloud-shared/reference; salmon --no-version-check quant -i /gcloud-shared/reference\n",
      "        -l A -1 ${FASTQR1} -2 ${FASTQR2} --validateMappings --seqBias --gcBias --posBias\n",
      "        --threads $(nproc) -o /gcloud-shared/sample; gsutil -mq cp -r /gcloud-shared/sample\n",
      "        ${GS_OUTPATH}; ls -lhR /gcloud-shared\n",
      "      entrypoint: bash\n",
      "      imageUri: us.gcr.io/nih-nia-lng-cbg/nihnialngcbg-salmon:2020-1\n",
      "      mounts:\n",
      "      - disk: gcloud-shared\n",
      "        path: /gcloud-shared\n",
      "    - alwaysRun: true\n",
      "      commands:\n",
      "      - /bin/sh\n",
      "      - -c\n",
      "      - gsutil -m -q cp /google/logs/output gs://nihnialngcbg-eqtl/rosmap_test/logs/salmon/102_120418\n",
      "      imageUri: google/cloud-sdk:slim\n",
      "    environment:\n",
      "      FASTQR1: /gcloud-shared/inputR1.gz\n",
      "      FASTQR2: /gcloud-shared/inputR2.gz\n",
      "      GS_OUTPATH: gs://nihnialngcbg-eqtl/rosmap_test/quants/gencode_r32/102_120418\n",
      "      SALMONTRANSCRIPTMODELS: gs://nihnialngcbg-eqtl/resources/references/salmon/gencode_32\n",
      "    resources:\n",
      "      virtualMachine:\n",
      "        bootDiskSizeGb: 10\n",
      "        bootImage: projects/cos-cloud/global/images/family/cos-stable\n",
      "        disks:\n",
      "        - name: gcloud-shared\n",
      "          sizeGb: 500\n",
      "        labels:\n",
      "          goog-pipelines-worker: 'true'\n",
      "        machineType: n1-standard-4\n",
      "        serviceAccount:\n",
      "          email: default\n",
      "          scopes:\n",
      "          - https://www.googleapis.com/auth/cloud-platform\n",
      "      zones:\n",
      "      - us-central1-f\n",
      "    timeout: 604800s\n",
      "  startTime: '2020-01-09T22:54:34.541001915Z'\n",
      "name: projects/463418702553/locations/us-central1/operations/8359902396683357875\n",
      "response:\n",
      "  '@type': type.googleapis.com/cloud.lifesciences.pipelines.RunPipelineResponse\n"
     ]
    }
   ],
   "source": [
    "#check job status, id OPID above\n",
    "op_id = 8359902396683357875\n",
    "\n",
    "!gcloud beta lifesciences operations describe {op_id}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
